{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "913d7735",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import anndata as ad\n",
    "import scanpy as sc\n",
    "import h5py\n",
    "from glob import glob\n",
    "\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import rcParams\n",
    "\n",
    "pd.options.display.max_columns = None\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59ec73e5",
   "metadata": {},
   "source": [
    "# Meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "009fb630",
   "metadata": {},
   "outputs": [],
   "source": [
    "ST_meta = pd.read_csv('../Dataset/C2L/ST/E-MTAB-11114.sdrf.txt', sep = '\\t')\n",
    "SN_meta = pd.read_csv('../Dataset/C2L/SN/E-MTAB-11115.sdrf.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e1a4fe47",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source Name</th>\n",
       "      <th>Characteristics[organism]</th>\n",
       "      <th>Characteristics[strain]</th>\n",
       "      <th>Characteristics[age]</th>\n",
       "      <th>Unit[time unit]</th>\n",
       "      <th>Term Source REF</th>\n",
       "      <th>Term Accession Number</th>\n",
       "      <th>Characteristics[developmental stage]</th>\n",
       "      <th>Characteristics[sex]</th>\n",
       "      <th>Characteristics[genotype]</th>\n",
       "      <th>Characteristics[organism part]</th>\n",
       "      <th>Characteristics[individual]</th>\n",
       "      <th>Material Type</th>\n",
       "      <th>Protocol REF</th>\n",
       "      <th>Performer</th>\n",
       "      <th>Protocol REF.1</th>\n",
       "      <th>Performer.1</th>\n",
       "      <th>Protocol REF.2</th>\n",
       "      <th>Performer.2</th>\n",
       "      <th>Extract Name</th>\n",
       "      <th>Comment[LIBRARY_LAYOUT]</th>\n",
       "      <th>Comment[LIBRARY_SELECTION]</th>\n",
       "      <th>Comment[LIBRARY_SOURCE]</th>\n",
       "      <th>Comment[LIBRARY_STRAND]</th>\n",
       "      <th>Comment[LIBRARY_STRATEGY]</th>\n",
       "      <th>Comment[NOMINAL_LENGTH]</th>\n",
       "      <th>Comment[NOMINAL_SDEV]</th>\n",
       "      <th>Comment[ORIENTATION]</th>\n",
       "      <th>Comment[end bias]</th>\n",
       "      <th>Comment[input molecule]</th>\n",
       "      <th>Comment[library construction]</th>\n",
       "      <th>Comment[primer]</th>\n",
       "      <th>Comment[single cell isolation]</th>\n",
       "      <th>Comment[spike in]</th>\n",
       "      <th>Protocol REF.3</th>\n",
       "      <th>Performer.3</th>\n",
       "      <th>Assay Name</th>\n",
       "      <th>Technology Type</th>\n",
       "      <th>Array Data File</th>\n",
       "      <th>Comment[MD5]</th>\n",
       "      <th>Comment[READ_TYPE]</th>\n",
       "      <th>Comment[READ_INDEX]</th>\n",
       "      <th>Comment[ArrayExpress FTP file]</th>\n",
       "      <th>Protocol REF.4</th>\n",
       "      <th>Performer.4</th>\n",
       "      <th>Protocol REF.5</th>\n",
       "      <th>Derived Array Data File</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file]</th>\n",
       "      <th>Protocol REF.6</th>\n",
       "      <th>Performer.5</th>\n",
       "      <th>Protocol REF.7</th>\n",
       "      <th>Derived Array Data File.1</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].1</th>\n",
       "      <th>Protocol REF.8</th>\n",
       "      <th>Performer.6</th>\n",
       "      <th>Protocol REF.9</th>\n",
       "      <th>Derived Array Data File.2</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].2</th>\n",
       "      <th>Protocol REF.10</th>\n",
       "      <th>Performer.7</th>\n",
       "      <th>Protocol REF.11</th>\n",
       "      <th>Derived Array Data File.3</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].3</th>\n",
       "      <th>Protocol REF.12</th>\n",
       "      <th>Performer.8</th>\n",
       "      <th>Protocol REF.13</th>\n",
       "      <th>Derived Array Data File.4</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].4</th>\n",
       "      <th>Protocol REF.14</th>\n",
       "      <th>Performer.9</th>\n",
       "      <th>Protocol REF.15</th>\n",
       "      <th>Derived Array Data File.5</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].5</th>\n",
       "      <th>Protocol REF.16</th>\n",
       "      <th>Performer.10</th>\n",
       "      <th>Protocol REF.17</th>\n",
       "      <th>Derived Array Data File.6</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].6</th>\n",
       "      <th>Factor Value[individual]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ST8059048</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059048</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059048</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059048_I1.fastq.gz</td>\n",
       "      <td>64ef7fd5012b7eabc33c549e798739ea</td>\n",
       "      <td>sample_barcode</td>\n",
       "      <td>index1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ST8059048</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059048</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059048</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059048_R1.fastq.gz</td>\n",
       "      <td>12d368cc4271a272f29f308ad09c65a7</td>\n",
       "      <td>spatial_barcode; umi_barcode</td>\n",
       "      <td>read1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ST8059048</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059048</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059048</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059048_R2.fastq.gz</td>\n",
       "      <td>e61f594765c8a9776a353fecd5fb8610</td>\n",
       "      <td>single</td>\n",
       "      <td>read2</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059048_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ST8059049</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059049</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059049</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059049_I1.fastq.gz</td>\n",
       "      <td>0df3b99e9d934ad0fc1106455a6b4361</td>\n",
       "      <td>sample_barcode</td>\n",
       "      <td>index1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ST8059049</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059049</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059049</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059049_R1.fastq.gz</td>\n",
       "      <td>85f7a09cffebd7a62318e7d9680b79f5</td>\n",
       "      <td>spatial_barcode; umi_barcode</td>\n",
       "      <td>read1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ST8059049</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059049</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059049</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059049_R2.fastq.gz</td>\n",
       "      <td>a3c56688a5bdebd81d870a39b9dd63ab</td>\n",
       "      <td>single</td>\n",
       "      <td>read2</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059049_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ST8059050</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059050</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059050</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059050_I1.fastq.gz</td>\n",
       "      <td>1db22a07bef761554650a729316715b9</td>\n",
       "      <td>sample_barcode</td>\n",
       "      <td>index1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ST8059050</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059050</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059050</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059050_R1.fastq.gz</td>\n",
       "      <td>a451aea3ca729322414c0be93330ece0</td>\n",
       "      <td>spatial_barcode; umi_barcode</td>\n",
       "      <td>read1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ST8059050</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059050</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059050</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059050_R2.fastq.gz</td>\n",
       "      <td>be46217e8be157a28fbf55f566a2cde5</td>\n",
       "      <td>single</td>\n",
       "      <td>read2</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059050_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ST8059051</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059051</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059051</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059051_I1.fastq.gz</td>\n",
       "      <td>6cc88612b82d61d561eb31eeacf34c84</td>\n",
       "      <td>sample_barcode</td>\n",
       "      <td>index1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ST8059051</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059051</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059051</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059051_R1.fastq.gz</td>\n",
       "      <td>b6563e05eeada5b0fce59b15123bc1ee</td>\n",
       "      <td>spatial_barcode; umi_barcode</td>\n",
       "      <td>read1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ST8059051</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059051</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059051</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059051_R2.fastq.gz</td>\n",
       "      <td>4e8230c5dca4fb9a664f3a20e495e40d</td>\n",
       "      <td>single</td>\n",
       "      <td>read2</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059051_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ST8059052</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059052</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059052</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059052_I1.fastq.gz</td>\n",
       "      <td>b92793d67517585133d335945b9fa86f</td>\n",
       "      <td>sample_barcode</td>\n",
       "      <td>index1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ST8059052</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059052</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059052</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059052_R1.fastq.gz</td>\n",
       "      <td>c95fb8f25246ec4e8388b0c1ed665645</td>\n",
       "      <td>spatial_barcode; umi_barcode</td>\n",
       "      <td>read1</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ST8059052</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>organism part</td>\n",
       "      <td>P-MTAB-115974</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115975</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115976</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059052</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10x Visium</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>other</td>\n",
       "      <td>none</td>\n",
       "      <td>P-MTAB-115977</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>ST8059052</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ST8059052_R2.fastq.gz</td>\n",
       "      <td>8639c1fa6d16a2e9d1953cc0892b7a61</td>\n",
       "      <td>single</td>\n",
       "      <td>read2</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_cloupe.cloupe</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_spatial.tar.gz</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115978</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115979</td>\n",
       "      <td>ST8059052_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Source Name Characteristics[organism] Characteristics[strain]  \\\n",
       "0    ST8059048              Mus musculus                 C57BL/6   \n",
       "1    ST8059048              Mus musculus                 C57BL/6   \n",
       "2    ST8059048              Mus musculus                 C57BL/6   \n",
       "3    ST8059049              Mus musculus                 C57BL/6   \n",
       "4    ST8059049              Mus musculus                 C57BL/6   \n",
       "5    ST8059049              Mus musculus                 C57BL/6   \n",
       "6    ST8059050              Mus musculus                 C57BL/6   \n",
       "7    ST8059050              Mus musculus                 C57BL/6   \n",
       "8    ST8059050              Mus musculus                 C57BL/6   \n",
       "9    ST8059051              Mus musculus                 C57BL/6   \n",
       "10   ST8059051              Mus musculus                 C57BL/6   \n",
       "11   ST8059051              Mus musculus                 C57BL/6   \n",
       "12   ST8059052              Mus musculus                 C57BL/6   \n",
       "13   ST8059052              Mus musculus                 C57BL/6   \n",
       "14   ST8059052              Mus musculus                 C57BL/6   \n",
       "\n",
       "    Characteristics[age] Unit[time unit] Term Source REF  \\\n",
       "0                     56             day             EFO   \n",
       "1                     56             day             EFO   \n",
       "2                     56             day             EFO   \n",
       "3                     56             day             EFO   \n",
       "4                     56             day             EFO   \n",
       "5                     56             day             EFO   \n",
       "6                     56             day             EFO   \n",
       "7                     56             day             EFO   \n",
       "8                     56             day             EFO   \n",
       "9                     56             day             EFO   \n",
       "10                    56             day             EFO   \n",
       "11                    56             day             EFO   \n",
       "12                    56             day             EFO   \n",
       "13                    56             day             EFO   \n",
       "14                    56             day             EFO   \n",
       "\n",
       "   Term Accession Number Characteristics[developmental stage]  \\\n",
       "0             UO_0000033                                adult   \n",
       "1             UO_0000033                                adult   \n",
       "2             UO_0000033                                adult   \n",
       "3             UO_0000033                                adult   \n",
       "4             UO_0000033                                adult   \n",
       "5             UO_0000033                                adult   \n",
       "6             UO_0000033                                adult   \n",
       "7             UO_0000033                                adult   \n",
       "8             UO_0000033                                adult   \n",
       "9             UO_0000033                                adult   \n",
       "10            UO_0000033                                adult   \n",
       "11            UO_0000033                                adult   \n",
       "12            UO_0000033                                adult   \n",
       "13            UO_0000033                                adult   \n",
       "14            UO_0000033                                adult   \n",
       "\n",
       "   Characteristics[sex] Characteristics[genotype]  \\\n",
       "0                female        wild type genotype   \n",
       "1                female        wild type genotype   \n",
       "2                female        wild type genotype   \n",
       "3                female        wild type genotype   \n",
       "4                female        wild type genotype   \n",
       "5                female        wild type genotype   \n",
       "6                female        wild type genotype   \n",
       "7                female        wild type genotype   \n",
       "8                female        wild type genotype   \n",
       "9                  male        wild type genotype   \n",
       "10                 male        wild type genotype   \n",
       "11                 male        wild type genotype   \n",
       "12                 male        wild type genotype   \n",
       "13                 male        wild type genotype   \n",
       "14                 male        wild type genotype   \n",
       "\n",
       "   Characteristics[organism part] Characteristics[individual]  Material Type  \\\n",
       "0                           brain                Individual 1  organism part   \n",
       "1                           brain                Individual 1  organism part   \n",
       "2                           brain                Individual 1  organism part   \n",
       "3                           brain                Individual 1  organism part   \n",
       "4                           brain                Individual 1  organism part   \n",
       "5                           brain                Individual 1  organism part   \n",
       "6                           brain                Individual 1  organism part   \n",
       "7                           brain                Individual 1  organism part   \n",
       "8                           brain                Individual 1  organism part   \n",
       "9                           brain                Individual 2  organism part   \n",
       "10                          brain                Individual 2  organism part   \n",
       "11                          brain                Individual 2  organism part   \n",
       "12                          brain                Individual 2  organism part   \n",
       "13                          brain                Individual 2  organism part   \n",
       "14                          brain                Individual 2  organism part   \n",
       "\n",
       "     Protocol REF                  Performer Protocol REF.1  \\\n",
       "0   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "1   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "2   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "3   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "4   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "5   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "6   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "7   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "8   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "9   P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "10  P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "11  P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "12  P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "13  P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "14  P-MTAB-115974  Wellcome Sanger Institute  P-MTAB-115975   \n",
       "\n",
       "                  Performer.1 Protocol REF.2                Performer.2  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115976  Wellcome Sanger Institute   \n",
       "\n",
       "   Extract Name Comment[LIBRARY_LAYOUT] Comment[LIBRARY_SELECTION]  \\\n",
       "0     ST8059048                  PAIRED                   Oligo-dT   \n",
       "1     ST8059048                  PAIRED                   Oligo-dT   \n",
       "2     ST8059048                  PAIRED                   Oligo-dT   \n",
       "3     ST8059049                  PAIRED                   Oligo-dT   \n",
       "4     ST8059049                  PAIRED                   Oligo-dT   \n",
       "5     ST8059049                  PAIRED                   Oligo-dT   \n",
       "6     ST8059050                  PAIRED                   Oligo-dT   \n",
       "7     ST8059050                  PAIRED                   Oligo-dT   \n",
       "8     ST8059050                  PAIRED                   Oligo-dT   \n",
       "9     ST8059051                  PAIRED                   Oligo-dT   \n",
       "10    ST8059051                  PAIRED                   Oligo-dT   \n",
       "11    ST8059051                  PAIRED                   Oligo-dT   \n",
       "12    ST8059052                  PAIRED                   Oligo-dT   \n",
       "13    ST8059052                  PAIRED                   Oligo-dT   \n",
       "14    ST8059052                  PAIRED                   Oligo-dT   \n",
       "\n",
       "       Comment[LIBRARY_SOURCE] Comment[LIBRARY_STRAND]  \\\n",
       "0   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "1   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "2   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "3   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "4   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "5   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "6   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "7   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "8   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "9   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "10  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "11  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "12  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "13  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "14  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "\n",
       "   Comment[LIBRARY_STRATEGY]  Comment[NOMINAL_LENGTH]  Comment[NOMINAL_SDEV]  \\\n",
       "0                      OTHER                      350                     50   \n",
       "1                      OTHER                      350                     50   \n",
       "2                      OTHER                      350                     50   \n",
       "3                      OTHER                      350                     50   \n",
       "4                      OTHER                      350                     50   \n",
       "5                      OTHER                      350                     50   \n",
       "6                      OTHER                      350                     50   \n",
       "7                      OTHER                      350                     50   \n",
       "8                      OTHER                      350                     50   \n",
       "9                      OTHER                      350                     50   \n",
       "10                     OTHER                      350                     50   \n",
       "11                     OTHER                      350                     50   \n",
       "12                     OTHER                      350                     50   \n",
       "13                     OTHER                      350                     50   \n",
       "14                     OTHER                      350                     50   \n",
       "\n",
       "   Comment[ORIENTATION] Comment[end bias] Comment[input molecule]  \\\n",
       "0           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "1           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "2           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "3           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "4           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "5           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "6           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "7           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "8           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "9           5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "10          5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "11          5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "12          5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "13          5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "14          5'-3'-3'-5'       3 prime tag               polyA RNA   \n",
       "\n",
       "   Comment[library construction] Comment[primer]  \\\n",
       "0                     10x Visium        oligo-dT   \n",
       "1                     10x Visium        oligo-dT   \n",
       "2                     10x Visium        oligo-dT   \n",
       "3                     10x Visium        oligo-dT   \n",
       "4                     10x Visium        oligo-dT   \n",
       "5                     10x Visium        oligo-dT   \n",
       "6                     10x Visium        oligo-dT   \n",
       "7                     10x Visium        oligo-dT   \n",
       "8                     10x Visium        oligo-dT   \n",
       "9                     10x Visium        oligo-dT   \n",
       "10                    10x Visium        oligo-dT   \n",
       "11                    10x Visium        oligo-dT   \n",
       "12                    10x Visium        oligo-dT   \n",
       "13                    10x Visium        oligo-dT   \n",
       "14                    10x Visium        oligo-dT   \n",
       "\n",
       "   Comment[single cell isolation] Comment[spike in] Protocol REF.3  \\\n",
       "0                           other              none  P-MTAB-115977   \n",
       "1                           other              none  P-MTAB-115977   \n",
       "2                           other              none  P-MTAB-115977   \n",
       "3                           other              none  P-MTAB-115977   \n",
       "4                           other              none  P-MTAB-115977   \n",
       "5                           other              none  P-MTAB-115977   \n",
       "6                           other              none  P-MTAB-115977   \n",
       "7                           other              none  P-MTAB-115977   \n",
       "8                           other              none  P-MTAB-115977   \n",
       "9                           other              none  P-MTAB-115977   \n",
       "10                          other              none  P-MTAB-115977   \n",
       "11                          other              none  P-MTAB-115977   \n",
       "12                          other              none  P-MTAB-115977   \n",
       "13                          other              none  P-MTAB-115977   \n",
       "14                          other              none  P-MTAB-115977   \n",
       "\n",
       "                  Performer.3 Assay Name   Technology Type  \\\n",
       "0   Wellcome Sanger Institute  ST8059048  sequencing assay   \n",
       "1   Wellcome Sanger Institute  ST8059048  sequencing assay   \n",
       "2   Wellcome Sanger Institute  ST8059048  sequencing assay   \n",
       "3   Wellcome Sanger Institute  ST8059049  sequencing assay   \n",
       "4   Wellcome Sanger Institute  ST8059049  sequencing assay   \n",
       "5   Wellcome Sanger Institute  ST8059049  sequencing assay   \n",
       "6   Wellcome Sanger Institute  ST8059050  sequencing assay   \n",
       "7   Wellcome Sanger Institute  ST8059050  sequencing assay   \n",
       "8   Wellcome Sanger Institute  ST8059050  sequencing assay   \n",
       "9   Wellcome Sanger Institute  ST8059051  sequencing assay   \n",
       "10  Wellcome Sanger Institute  ST8059051  sequencing assay   \n",
       "11  Wellcome Sanger Institute  ST8059051  sequencing assay   \n",
       "12  Wellcome Sanger Institute  ST8059052  sequencing assay   \n",
       "13  Wellcome Sanger Institute  ST8059052  sequencing assay   \n",
       "14  Wellcome Sanger Institute  ST8059052  sequencing assay   \n",
       "\n",
       "          Array Data File                      Comment[MD5]  \\\n",
       "0   ST8059048_I1.fastq.gz  64ef7fd5012b7eabc33c549e798739ea   \n",
       "1   ST8059048_R1.fastq.gz  12d368cc4271a272f29f308ad09c65a7   \n",
       "2   ST8059048_R2.fastq.gz  e61f594765c8a9776a353fecd5fb8610   \n",
       "3   ST8059049_I1.fastq.gz  0df3b99e9d934ad0fc1106455a6b4361   \n",
       "4   ST8059049_R1.fastq.gz  85f7a09cffebd7a62318e7d9680b79f5   \n",
       "5   ST8059049_R2.fastq.gz  a3c56688a5bdebd81d870a39b9dd63ab   \n",
       "6   ST8059050_I1.fastq.gz  1db22a07bef761554650a729316715b9   \n",
       "7   ST8059050_R1.fastq.gz  a451aea3ca729322414c0be93330ece0   \n",
       "8   ST8059050_R2.fastq.gz  be46217e8be157a28fbf55f566a2cde5   \n",
       "9   ST8059051_I1.fastq.gz  6cc88612b82d61d561eb31eeacf34c84   \n",
       "10  ST8059051_R1.fastq.gz  b6563e05eeada5b0fce59b15123bc1ee   \n",
       "11  ST8059051_R2.fastq.gz  4e8230c5dca4fb9a664f3a20e495e40d   \n",
       "12  ST8059052_I1.fastq.gz  b92793d67517585133d335945b9fa86f   \n",
       "13  ST8059052_R1.fastq.gz  c95fb8f25246ec4e8388b0c1ed665645   \n",
       "14  ST8059052_R2.fastq.gz  8639c1fa6d16a2e9d1953cc0892b7a61   \n",
       "\n",
       "              Comment[READ_TYPE] Comment[READ_INDEX]  \\\n",
       "0                 sample_barcode              index1   \n",
       "1   spatial_barcode; umi_barcode               read1   \n",
       "2                         single               read2   \n",
       "3                 sample_barcode              index1   \n",
       "4   spatial_barcode; umi_barcode               read1   \n",
       "5                         single               read2   \n",
       "6                 sample_barcode              index1   \n",
       "7   spatial_barcode; umi_barcode               read1   \n",
       "8                         single               read2   \n",
       "9                 sample_barcode              index1   \n",
       "10  spatial_barcode; umi_barcode               read1   \n",
       "11                        single               read2   \n",
       "12                sample_barcode              index1   \n",
       "13  spatial_barcode; umi_barcode               read1   \n",
       "14                        single               read2   \n",
       "\n",
       "                       Comment[ArrayExpress FTP file] Protocol REF.4  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "\n",
       "                  Performer.4 Protocol REF.5  Derived Array Data File  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115979  ST8059048_cloupe.cloupe   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115979  ST8059048_cloupe.cloupe   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115979  ST8059048_cloupe.cloupe   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115979  ST8059049_cloupe.cloupe   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115979  ST8059049_cloupe.cloupe   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115979  ST8059049_cloupe.cloupe   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115979  ST8059050_cloupe.cloupe   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115979  ST8059050_cloupe.cloupe   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115979  ST8059050_cloupe.cloupe   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115979  ST8059051_cloupe.cloupe   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115979  ST8059051_cloupe.cloupe   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115979  ST8059051_cloupe.cloupe   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115979  ST8059052_cloupe.cloupe   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115979  ST8059052_cloupe.cloupe   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115979  ST8059052_cloupe.cloupe   \n",
       "\n",
       "              Comment [Derived ArrayExpress FTP file] Protocol REF.6  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "\n",
       "                  Performer.5 Protocol REF.7  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115979   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115979   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115979   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115979   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115979   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115979   \n",
       "\n",
       "                  Derived Array Data File.1  \\\n",
       "0   ST8059048_filtered_feature_bc_matrix.h5   \n",
       "1   ST8059048_filtered_feature_bc_matrix.h5   \n",
       "2   ST8059048_filtered_feature_bc_matrix.h5   \n",
       "3   ST8059049_filtered_feature_bc_matrix.h5   \n",
       "4   ST8059049_filtered_feature_bc_matrix.h5   \n",
       "5   ST8059049_filtered_feature_bc_matrix.h5   \n",
       "6   ST8059050_filtered_feature_bc_matrix.h5   \n",
       "7   ST8059050_filtered_feature_bc_matrix.h5   \n",
       "8   ST8059050_filtered_feature_bc_matrix.h5   \n",
       "9   ST8059051_filtered_feature_bc_matrix.h5   \n",
       "10  ST8059051_filtered_feature_bc_matrix.h5   \n",
       "11  ST8059051_filtered_feature_bc_matrix.h5   \n",
       "12  ST8059052_filtered_feature_bc_matrix.h5   \n",
       "13  ST8059052_filtered_feature_bc_matrix.h5   \n",
       "14  ST8059052_filtered_feature_bc_matrix.h5   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].1 Protocol REF.8  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115978   \n",
       "\n",
       "                  Performer.6 Protocol REF.9      Derived Array Data File.2  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115979  ST8059048_metrics_summary.csv   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115979  ST8059048_metrics_summary.csv   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115979  ST8059048_metrics_summary.csv   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115979  ST8059049_metrics_summary.csv   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115979  ST8059049_metrics_summary.csv   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115979  ST8059049_metrics_summary.csv   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115979  ST8059050_metrics_summary.csv   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115979  ST8059050_metrics_summary.csv   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115979  ST8059050_metrics_summary.csv   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115979  ST8059051_metrics_summary.csv   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115979  ST8059051_metrics_summary.csv   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115979  ST8059051_metrics_summary.csv   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115979  ST8059052_metrics_summary.csv   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115979  ST8059052_metrics_summary.csv   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115979  ST8059052_metrics_summary.csv   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].2 Protocol REF.10  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "\n",
       "                  Performer.7 Protocol REF.11   Derived Array Data File.3  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_molecule_info.h5   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_molecule_info.h5   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_molecule_info.h5   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_molecule_info.h5   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_molecule_info.h5   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_molecule_info.h5   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_molecule_info.h5   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_molecule_info.h5   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_molecule_info.h5   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115979  ST8059051_molecule_info.h5   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115979  ST8059051_molecule_info.h5   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115979  ST8059051_molecule_info.h5   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_molecule_info.h5   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_molecule_info.h5   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_molecule_info.h5   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].3 Protocol REF.12  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "\n",
       "                  Performer.8 Protocol REF.13  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115979   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115979   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115979   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115979   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115979   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115979   \n",
       "\n",
       "             Derived Array Data File.4  \\\n",
       "0   ST8059048_raw_feature_bc_matrix.h5   \n",
       "1   ST8059048_raw_feature_bc_matrix.h5   \n",
       "2   ST8059048_raw_feature_bc_matrix.h5   \n",
       "3   ST8059049_raw_feature_bc_matrix.h5   \n",
       "4   ST8059049_raw_feature_bc_matrix.h5   \n",
       "5   ST8059049_raw_feature_bc_matrix.h5   \n",
       "6   ST8059050_raw_feature_bc_matrix.h5   \n",
       "7   ST8059050_raw_feature_bc_matrix.h5   \n",
       "8   ST8059050_raw_feature_bc_matrix.h5   \n",
       "9   ST8059051_raw_feature_bc_matrix.h5   \n",
       "10  ST8059051_raw_feature_bc_matrix.h5   \n",
       "11  ST8059051_raw_feature_bc_matrix.h5   \n",
       "12  ST8059052_raw_feature_bc_matrix.h5   \n",
       "13  ST8059052_raw_feature_bc_matrix.h5   \n",
       "14  ST8059052_raw_feature_bc_matrix.h5   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].4 Protocol REF.14  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "\n",
       "                  Performer.9 Protocol REF.15 Derived Array Data File.5  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_spatial.tar.gz   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_spatial.tar.gz   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_spatial.tar.gz   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_spatial.tar.gz   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_spatial.tar.gz   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_spatial.tar.gz   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_spatial.tar.gz   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_spatial.tar.gz   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_spatial.tar.gz   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115979  ST8059051_spatial.tar.gz   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115979  ST8059051_spatial.tar.gz   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115979  ST8059051_spatial.tar.gz   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_spatial.tar.gz   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_spatial.tar.gz   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_spatial.tar.gz   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].5 Protocol REF.16  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115978   \n",
       "\n",
       "                 Performer.10 Protocol REF.17   Derived Array Data File.6  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_web_summary.html   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_web_summary.html   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115979  ST8059048_web_summary.html   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_web_summary.html   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_web_summary.html   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115979  ST8059049_web_summary.html   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_web_summary.html   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_web_summary.html   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115979  ST8059050_web_summary.html   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115979  ST8059051_web_summary.html   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115979  ST8059051_web_summary.html   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115979  ST8059051_web_summary.html   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_web_summary.html   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_web_summary.html   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115979  ST8059052_web_summary.html   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].6 Factor Value[individual]  \n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ST_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3664327e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source Name</th>\n",
       "      <th>Comment[ENA_SAMPLE]</th>\n",
       "      <th>Comment[BioSD_SAMPLE]</th>\n",
       "      <th>Characteristics[organism]</th>\n",
       "      <th>Characteristics[strain]</th>\n",
       "      <th>Characteristics[age]</th>\n",
       "      <th>Unit[time unit]</th>\n",
       "      <th>Term Source REF</th>\n",
       "      <th>Term Accession Number</th>\n",
       "      <th>Characteristics[developmental stage]</th>\n",
       "      <th>Characteristics[sex]</th>\n",
       "      <th>Characteristics[genotype]</th>\n",
       "      <th>Characteristics[organism part]</th>\n",
       "      <th>Characteristics[individual]</th>\n",
       "      <th>Characteristics[cell type]</th>\n",
       "      <th>Material Type</th>\n",
       "      <th>Protocol REF</th>\n",
       "      <th>Performer</th>\n",
       "      <th>Protocol REF.1</th>\n",
       "      <th>Performer.1</th>\n",
       "      <th>Protocol REF.2</th>\n",
       "      <th>Performer.2</th>\n",
       "      <th>Extract Name</th>\n",
       "      <th>Comment[LIBRARY_LAYOUT]</th>\n",
       "      <th>Comment[LIBRARY_SELECTION]</th>\n",
       "      <th>Comment[LIBRARY_SOURCE]</th>\n",
       "      <th>Comment[LIBRARY_STRAND]</th>\n",
       "      <th>Comment[LIBRARY_STRATEGY]</th>\n",
       "      <th>Comment[NOMINAL_LENGTH]</th>\n",
       "      <th>Comment[NOMINAL_SDEV]</th>\n",
       "      <th>Comment[ORIENTATION]</th>\n",
       "      <th>Comment[cdna read]</th>\n",
       "      <th>Comment[cdna read offset]</th>\n",
       "      <th>Comment[cdna read size]</th>\n",
       "      <th>Comment[cell barcode offset]</th>\n",
       "      <th>Comment[cell barcode read]</th>\n",
       "      <th>Comment[cell barcode size]</th>\n",
       "      <th>Comment[end bias]</th>\n",
       "      <th>Comment[input molecule]</th>\n",
       "      <th>Comment[library construction]</th>\n",
       "      <th>Comment[primer]</th>\n",
       "      <th>Comment[sample barcode offset]</th>\n",
       "      <th>Comment[sample barcode read]</th>\n",
       "      <th>Comment[sample barcode size]</th>\n",
       "      <th>Comment[single cell isolation]</th>\n",
       "      <th>Comment[spike in]</th>\n",
       "      <th>Comment[umi barcode offset]</th>\n",
       "      <th>Comment[umi barcode read]</th>\n",
       "      <th>Comment[umi barcode size]</th>\n",
       "      <th>Protocol REF.3</th>\n",
       "      <th>Performer.3</th>\n",
       "      <th>Assay Name</th>\n",
       "      <th>Technology Type</th>\n",
       "      <th>Comment[ENA_EXPERIMENT]</th>\n",
       "      <th>Scan Name</th>\n",
       "      <th>Comment[SUBMITTED_FILE_NAME]</th>\n",
       "      <th>Comment[ENA_RUN]</th>\n",
       "      <th>Comment[FASTQ_URI]</th>\n",
       "      <th>Comment[read1 file]</th>\n",
       "      <th>Comment[read2 file]</th>\n",
       "      <th>Comment[index1 file]</th>\n",
       "      <th>Protocol REF.4</th>\n",
       "      <th>Performer.4</th>\n",
       "      <th>Protocol REF.5</th>\n",
       "      <th>Derived Array Data File</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file]</th>\n",
       "      <th>Protocol REF.6</th>\n",
       "      <th>Performer.5</th>\n",
       "      <th>Protocol REF.7</th>\n",
       "      <th>Derived Array Data File.1</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].1</th>\n",
       "      <th>Protocol REF.8</th>\n",
       "      <th>Performer.6</th>\n",
       "      <th>Protocol REF.9</th>\n",
       "      <th>Derived Array Data File.2</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].2</th>\n",
       "      <th>Protocol REF.10</th>\n",
       "      <th>Performer.7</th>\n",
       "      <th>Protocol REF.11</th>\n",
       "      <th>Derived Array Data File.3</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].3</th>\n",
       "      <th>Protocol REF.12</th>\n",
       "      <th>Performer.8</th>\n",
       "      <th>Protocol REF.13</th>\n",
       "      <th>Derived Array Data File.4</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].4</th>\n",
       "      <th>Protocol REF.14</th>\n",
       "      <th>Performer.9</th>\n",
       "      <th>Protocol REF.15</th>\n",
       "      <th>Derived Array Data File.5</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file].5</th>\n",
       "      <th>Factor Value[individual]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>ERS8488760</td>\n",
       "      <td>SAMEA10837525</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821446</td>\n",
       "      <td>5705STDY8058280_I1.fastq.gz</td>\n",
       "      <td>5705STDY8058280_I1.fastq.gz</td>\n",
       "      <td>ERR7251990</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058280_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058280_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058280_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>ERS8488760</td>\n",
       "      <td>SAMEA10837525</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821446</td>\n",
       "      <td>5705STDY8058280_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058280_R1.fastq.gz</td>\n",
       "      <td>ERR7251990</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058280_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058280_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058280_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>ERS8488760</td>\n",
       "      <td>SAMEA10837525</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058280</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821446</td>\n",
       "      <td>5705STDY8058280_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058280_R2.fastq.gz</td>\n",
       "      <td>ERR7251990</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058280_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058280_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058280_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058280_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>ERS8488761</td>\n",
       "      <td>SAMEA10837526</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821447</td>\n",
       "      <td>5705STDY8058281_I1.fastq.gz</td>\n",
       "      <td>5705STDY8058281_I1.fastq.gz</td>\n",
       "      <td>ERR7251991</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058281_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058281_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058281_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>ERS8488761</td>\n",
       "      <td>SAMEA10837526</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821447</td>\n",
       "      <td>5705STDY8058281_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058281_R1.fastq.gz</td>\n",
       "      <td>ERR7251991</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058281_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058281_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058281_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>ERS8488761</td>\n",
       "      <td>SAMEA10837526</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058281</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821447</td>\n",
       "      <td>5705STDY8058281_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058281_R2.fastq.gz</td>\n",
       "      <td>ERR7251991</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058281_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058281_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058281_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058281_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>ERS8488762</td>\n",
       "      <td>SAMEA10837527</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821448</td>\n",
       "      <td>5705STDY8058282_I1.fastq.gz</td>\n",
       "      <td>5705STDY8058282_I1.fastq.gz</td>\n",
       "      <td>ERR7251992</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058282_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058282_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058282_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>ERS8488762</td>\n",
       "      <td>SAMEA10837527</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821448</td>\n",
       "      <td>5705STDY8058282_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058282_R1.fastq.gz</td>\n",
       "      <td>ERR7251992</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058282_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058282_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058282_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>ERS8488762</td>\n",
       "      <td>SAMEA10837527</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 1</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058282</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821448</td>\n",
       "      <td>5705STDY8058282_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058282_R2.fastq.gz</td>\n",
       "      <td>ERR7251992</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058282_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058282_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058282_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058282_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>ERS8488763</td>\n",
       "      <td>SAMEA10837528</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821449</td>\n",
       "      <td>5705STDY8058283_I1.fastq.gz</td>\n",
       "      <td>5705STDY8058283_I1.fastq.gz</td>\n",
       "      <td>ERR7251993</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058283_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058283_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058283_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>ERS8488763</td>\n",
       "      <td>SAMEA10837528</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821449</td>\n",
       "      <td>5705STDY8058283_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058283_R1.fastq.gz</td>\n",
       "      <td>ERR7251993</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058283_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058283_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058283_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>ERS8488763</td>\n",
       "      <td>SAMEA10837528</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058283</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821449</td>\n",
       "      <td>5705STDY8058283_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058283_R2.fastq.gz</td>\n",
       "      <td>ERR7251993</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058283_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058283_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058283_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058283_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>ERS8488764</td>\n",
       "      <td>SAMEA10837529</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821450</td>\n",
       "      <td>5705STDY8058284_I1.fastq.gz</td>\n",
       "      <td>5705STDY8058284_I1.fastq.gz</td>\n",
       "      <td>ERR7251994</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058284_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058284_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058284_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>ERS8488764</td>\n",
       "      <td>SAMEA10837529</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821450</td>\n",
       "      <td>5705STDY8058284_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058284_R1.fastq.gz</td>\n",
       "      <td>ERR7251994</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058284_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058284_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058284_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>ERS8488764</td>\n",
       "      <td>SAMEA10837529</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821450</td>\n",
       "      <td>5705STDY8058284_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058284_R2.fastq.gz</td>\n",
       "      <td>ERR7251994</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058284_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058284_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058284_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058284_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>ERS8488765</td>\n",
       "      <td>SAMEA10837530</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821451</td>\n",
       "      <td>5705STDY8058285_I1.fastq.gz</td>\n",
       "      <td>5705STDY8058285_I1.fastq.gz</td>\n",
       "      <td>ERR7251995</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058285_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058285_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058285_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>ERS8488765</td>\n",
       "      <td>SAMEA10837530</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821451</td>\n",
       "      <td>5705STDY8058285_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058285_R1.fastq.gz</td>\n",
       "      <td>ERR7251995</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058285_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058285_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058285_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>ERS8488765</td>\n",
       "      <td>SAMEA10837530</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6</td>\n",
       "      <td>56</td>\n",
       "      <td>day</td>\n",
       "      <td>EFO</td>\n",
       "      <td>UO_0000033</td>\n",
       "      <td>adult</td>\n",
       "      <td>male</td>\n",
       "      <td>wild type genotype</td>\n",
       "      <td>brain</td>\n",
       "      <td>Individual 2</td>\n",
       "      <td>Single nuclei dissociated from frozen tissue.</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>P-MTAB-115980</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115981</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115982</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>Oligo-dT</td>\n",
       "      <td>TRANSCRIPTOMIC SINGLE CELL</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>350</td>\n",
       "      <td>50</td>\n",
       "      <td>5'-3'-3'-5'</td>\n",
       "      <td>read 2</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>read 1</td>\n",
       "      <td>16</td>\n",
       "      <td>3 prime tag</td>\n",
       "      <td>polyA RNA</td>\n",
       "      <td>10xV3</td>\n",
       "      <td>oligo-dT</td>\n",
       "      <td>0</td>\n",
       "      <td>index 1</td>\n",
       "      <td>8</td>\n",
       "      <td>10x</td>\n",
       "      <td>none</td>\n",
       "      <td>16</td>\n",
       "      <td>read 1</td>\n",
       "      <td>12</td>\n",
       "      <td>P-MTAB-115983</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>sequencing assay</td>\n",
       "      <td>ERX6821451</td>\n",
       "      <td>5705STDY8058285_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058285_R2.fastq.gz</td>\n",
       "      <td>ERR7251995</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...</td>\n",
       "      <td>5705STDY8058285_R1.fastq.gz</td>\n",
       "      <td>5705STDY8058285_R2.fastq.gz</td>\n",
       "      <td>5705STDY8058285_I1.fastq.gz</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_filtered_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_molecule_info.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_metrics_summary.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_raw_feature_bc_matrix.h5</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>5705STDY8058285_web_summary.html</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>P-MTAB-115984</td>\n",
       "      <td>Wellcome Sanger Institute</td>\n",
       "      <td>P-MTAB-115985</td>\n",
       "      <td>cell_annotation.csv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>Individual 2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Source Name Comment[ENA_SAMPLE] Comment[BioSD_SAMPLE]  \\\n",
       "0   5705STDY8058280          ERS8488760         SAMEA10837525   \n",
       "1   5705STDY8058280          ERS8488760         SAMEA10837525   \n",
       "2   5705STDY8058280          ERS8488760         SAMEA10837525   \n",
       "3   5705STDY8058281          ERS8488761         SAMEA10837526   \n",
       "4   5705STDY8058281          ERS8488761         SAMEA10837526   \n",
       "5   5705STDY8058281          ERS8488761         SAMEA10837526   \n",
       "6   5705STDY8058282          ERS8488762         SAMEA10837527   \n",
       "7   5705STDY8058282          ERS8488762         SAMEA10837527   \n",
       "8   5705STDY8058282          ERS8488762         SAMEA10837527   \n",
       "9   5705STDY8058283          ERS8488763         SAMEA10837528   \n",
       "10  5705STDY8058283          ERS8488763         SAMEA10837528   \n",
       "11  5705STDY8058283          ERS8488763         SAMEA10837528   \n",
       "12  5705STDY8058284          ERS8488764         SAMEA10837529   \n",
       "13  5705STDY8058284          ERS8488764         SAMEA10837529   \n",
       "14  5705STDY8058284          ERS8488764         SAMEA10837529   \n",
       "15  5705STDY8058285          ERS8488765         SAMEA10837530   \n",
       "16  5705STDY8058285          ERS8488765         SAMEA10837530   \n",
       "17  5705STDY8058285          ERS8488765         SAMEA10837530   \n",
       "\n",
       "   Characteristics[organism] Characteristics[strain]  Characteristics[age]  \\\n",
       "0               Mus musculus                 C57BL/6                    56   \n",
       "1               Mus musculus                 C57BL/6                    56   \n",
       "2               Mus musculus                 C57BL/6                    56   \n",
       "3               Mus musculus                 C57BL/6                    56   \n",
       "4               Mus musculus                 C57BL/6                    56   \n",
       "5               Mus musculus                 C57BL/6                    56   \n",
       "6               Mus musculus                 C57BL/6                    56   \n",
       "7               Mus musculus                 C57BL/6                    56   \n",
       "8               Mus musculus                 C57BL/6                    56   \n",
       "9               Mus musculus                 C57BL/6                    56   \n",
       "10              Mus musculus                 C57BL/6                    56   \n",
       "11              Mus musculus                 C57BL/6                    56   \n",
       "12              Mus musculus                 C57BL/6                    56   \n",
       "13              Mus musculus                 C57BL/6                    56   \n",
       "14              Mus musculus                 C57BL/6                    56   \n",
       "15              Mus musculus                 C57BL/6                    56   \n",
       "16              Mus musculus                 C57BL/6                    56   \n",
       "17              Mus musculus                 C57BL/6                    56   \n",
       "\n",
       "   Unit[time unit] Term Source REF Term Accession Number  \\\n",
       "0              day             EFO            UO_0000033   \n",
       "1              day             EFO            UO_0000033   \n",
       "2              day             EFO            UO_0000033   \n",
       "3              day             EFO            UO_0000033   \n",
       "4              day             EFO            UO_0000033   \n",
       "5              day             EFO            UO_0000033   \n",
       "6              day             EFO            UO_0000033   \n",
       "7              day             EFO            UO_0000033   \n",
       "8              day             EFO            UO_0000033   \n",
       "9              day             EFO            UO_0000033   \n",
       "10             day             EFO            UO_0000033   \n",
       "11             day             EFO            UO_0000033   \n",
       "12             day             EFO            UO_0000033   \n",
       "13             day             EFO            UO_0000033   \n",
       "14             day             EFO            UO_0000033   \n",
       "15             day             EFO            UO_0000033   \n",
       "16             day             EFO            UO_0000033   \n",
       "17             day             EFO            UO_0000033   \n",
       "\n",
       "   Characteristics[developmental stage] Characteristics[sex]  \\\n",
       "0                                 adult               female   \n",
       "1                                 adult               female   \n",
       "2                                 adult               female   \n",
       "3                                 adult               female   \n",
       "4                                 adult               female   \n",
       "5                                 adult               female   \n",
       "6                                 adult               female   \n",
       "7                                 adult               female   \n",
       "8                                 adult               female   \n",
       "9                                 adult                 male   \n",
       "10                                adult                 male   \n",
       "11                                adult                 male   \n",
       "12                                adult                 male   \n",
       "13                                adult                 male   \n",
       "14                                adult                 male   \n",
       "15                                adult                 male   \n",
       "16                                adult                 male   \n",
       "17                                adult                 male   \n",
       "\n",
       "   Characteristics[genotype] Characteristics[organism part]  \\\n",
       "0         wild type genotype                          brain   \n",
       "1         wild type genotype                          brain   \n",
       "2         wild type genotype                          brain   \n",
       "3         wild type genotype                          brain   \n",
       "4         wild type genotype                          brain   \n",
       "5         wild type genotype                          brain   \n",
       "6         wild type genotype                          brain   \n",
       "7         wild type genotype                          brain   \n",
       "8         wild type genotype                          brain   \n",
       "9         wild type genotype                          brain   \n",
       "10        wild type genotype                          brain   \n",
       "11        wild type genotype                          brain   \n",
       "12        wild type genotype                          brain   \n",
       "13        wild type genotype                          brain   \n",
       "14        wild type genotype                          brain   \n",
       "15        wild type genotype                          brain   \n",
       "16        wild type genotype                          brain   \n",
       "17        wild type genotype                          brain   \n",
       "\n",
       "   Characteristics[individual]                     Characteristics[cell type]  \\\n",
       "0                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "1                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "2                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "3                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "4                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "5                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "6                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "7                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "8                 Individual 1  Single nuclei dissociated from frozen tissue.   \n",
       "9                 Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "10                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "11                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "12                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "13                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "14                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "15                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "16                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "17                Individual 2  Single nuclei dissociated from frozen tissue.   \n",
       "\n",
       "   Material Type   Protocol REF                  Performer Protocol REF.1  \\\n",
       "0        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "1        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "2        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "3        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "4        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "5        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "6        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "7        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "8        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "9        nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "10       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "11       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "12       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "13       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "14       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "15       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "16       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "17       nucleus  P-MTAB-115980  Wellcome Sanger Institute  P-MTAB-115981   \n",
       "\n",
       "                  Performer.1 Protocol REF.2                Performer.2  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "15  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "16  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "17  Wellcome Sanger Institute  P-MTAB-115982  Wellcome Sanger Institute   \n",
       "\n",
       "       Extract Name Comment[LIBRARY_LAYOUT] Comment[LIBRARY_SELECTION]  \\\n",
       "0   5705STDY8058280                  PAIRED                   Oligo-dT   \n",
       "1   5705STDY8058280                  PAIRED                   Oligo-dT   \n",
       "2   5705STDY8058280                  PAIRED                   Oligo-dT   \n",
       "3   5705STDY8058281                  PAIRED                   Oligo-dT   \n",
       "4   5705STDY8058281                  PAIRED                   Oligo-dT   \n",
       "5   5705STDY8058281                  PAIRED                   Oligo-dT   \n",
       "6   5705STDY8058282                  PAIRED                   Oligo-dT   \n",
       "7   5705STDY8058282                  PAIRED                   Oligo-dT   \n",
       "8   5705STDY8058282                  PAIRED                   Oligo-dT   \n",
       "9   5705STDY8058283                  PAIRED                   Oligo-dT   \n",
       "10  5705STDY8058283                  PAIRED                   Oligo-dT   \n",
       "11  5705STDY8058283                  PAIRED                   Oligo-dT   \n",
       "12  5705STDY8058284                  PAIRED                   Oligo-dT   \n",
       "13  5705STDY8058284                  PAIRED                   Oligo-dT   \n",
       "14  5705STDY8058284                  PAIRED                   Oligo-dT   \n",
       "15  5705STDY8058285                  PAIRED                   Oligo-dT   \n",
       "16  5705STDY8058285                  PAIRED                   Oligo-dT   \n",
       "17  5705STDY8058285                  PAIRED                   Oligo-dT   \n",
       "\n",
       "       Comment[LIBRARY_SOURCE] Comment[LIBRARY_STRAND]  \\\n",
       "0   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "1   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "2   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "3   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "4   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "5   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "6   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "7   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "8   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "9   TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "10  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "11  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "12  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "13  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "14  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "15  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "16  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "17  TRANSCRIPTOMIC SINGLE CELL          not applicable   \n",
       "\n",
       "   Comment[LIBRARY_STRATEGY]  Comment[NOMINAL_LENGTH]  Comment[NOMINAL_SDEV]  \\\n",
       "0                    RNA-Seq                      350                     50   \n",
       "1                    RNA-Seq                      350                     50   \n",
       "2                    RNA-Seq                      350                     50   \n",
       "3                    RNA-Seq                      350                     50   \n",
       "4                    RNA-Seq                      350                     50   \n",
       "5                    RNA-Seq                      350                     50   \n",
       "6                    RNA-Seq                      350                     50   \n",
       "7                    RNA-Seq                      350                     50   \n",
       "8                    RNA-Seq                      350                     50   \n",
       "9                    RNA-Seq                      350                     50   \n",
       "10                   RNA-Seq                      350                     50   \n",
       "11                   RNA-Seq                      350                     50   \n",
       "12                   RNA-Seq                      350                     50   \n",
       "13                   RNA-Seq                      350                     50   \n",
       "14                   RNA-Seq                      350                     50   \n",
       "15                   RNA-Seq                      350                     50   \n",
       "16                   RNA-Seq                      350                     50   \n",
       "17                   RNA-Seq                      350                     50   \n",
       "\n",
       "   Comment[ORIENTATION] Comment[cdna read]  Comment[cdna read offset]  \\\n",
       "0           5'-3'-3'-5'             read 2                          0   \n",
       "1           5'-3'-3'-5'             read 2                          0   \n",
       "2           5'-3'-3'-5'             read 2                          0   \n",
       "3           5'-3'-3'-5'             read 2                          0   \n",
       "4           5'-3'-3'-5'             read 2                          0   \n",
       "5           5'-3'-3'-5'             read 2                          0   \n",
       "6           5'-3'-3'-5'             read 2                          0   \n",
       "7           5'-3'-3'-5'             read 2                          0   \n",
       "8           5'-3'-3'-5'             read 2                          0   \n",
       "9           5'-3'-3'-5'             read 2                          0   \n",
       "10          5'-3'-3'-5'             read 2                          0   \n",
       "11          5'-3'-3'-5'             read 2                          0   \n",
       "12          5'-3'-3'-5'             read 2                          0   \n",
       "13          5'-3'-3'-5'             read 2                          0   \n",
       "14          5'-3'-3'-5'             read 2                          0   \n",
       "15          5'-3'-3'-5'             read 2                          0   \n",
       "16          5'-3'-3'-5'             read 2                          0   \n",
       "17          5'-3'-3'-5'             read 2                          0   \n",
       "\n",
       "    Comment[cdna read size]  Comment[cell barcode offset]  \\\n",
       "0                        91                             0   \n",
       "1                        91                             0   \n",
       "2                        91                             0   \n",
       "3                        91                             0   \n",
       "4                        91                             0   \n",
       "5                        91                             0   \n",
       "6                        91                             0   \n",
       "7                        91                             0   \n",
       "8                        91                             0   \n",
       "9                        91                             0   \n",
       "10                       91                             0   \n",
       "11                       91                             0   \n",
       "12                       91                             0   \n",
       "13                       91                             0   \n",
       "14                       91                             0   \n",
       "15                       91                             0   \n",
       "16                       91                             0   \n",
       "17                       91                             0   \n",
       "\n",
       "   Comment[cell barcode read]  Comment[cell barcode size] Comment[end bias]  \\\n",
       "0                      read 1                          16       3 prime tag   \n",
       "1                      read 1                          16       3 prime tag   \n",
       "2                      read 1                          16       3 prime tag   \n",
       "3                      read 1                          16       3 prime tag   \n",
       "4                      read 1                          16       3 prime tag   \n",
       "5                      read 1                          16       3 prime tag   \n",
       "6                      read 1                          16       3 prime tag   \n",
       "7                      read 1                          16       3 prime tag   \n",
       "8                      read 1                          16       3 prime tag   \n",
       "9                      read 1                          16       3 prime tag   \n",
       "10                     read 1                          16       3 prime tag   \n",
       "11                     read 1                          16       3 prime tag   \n",
       "12                     read 1                          16       3 prime tag   \n",
       "13                     read 1                          16       3 prime tag   \n",
       "14                     read 1                          16       3 prime tag   \n",
       "15                     read 1                          16       3 prime tag   \n",
       "16                     read 1                          16       3 prime tag   \n",
       "17                     read 1                          16       3 prime tag   \n",
       "\n",
       "   Comment[input molecule] Comment[library construction] Comment[primer]  \\\n",
       "0                polyA RNA                         10xV3        oligo-dT   \n",
       "1                polyA RNA                         10xV3        oligo-dT   \n",
       "2                polyA RNA                         10xV3        oligo-dT   \n",
       "3                polyA RNA                         10xV3        oligo-dT   \n",
       "4                polyA RNA                         10xV3        oligo-dT   \n",
       "5                polyA RNA                         10xV3        oligo-dT   \n",
       "6                polyA RNA                         10xV3        oligo-dT   \n",
       "7                polyA RNA                         10xV3        oligo-dT   \n",
       "8                polyA RNA                         10xV3        oligo-dT   \n",
       "9                polyA RNA                         10xV3        oligo-dT   \n",
       "10               polyA RNA                         10xV3        oligo-dT   \n",
       "11               polyA RNA                         10xV3        oligo-dT   \n",
       "12               polyA RNA                         10xV3        oligo-dT   \n",
       "13               polyA RNA                         10xV3        oligo-dT   \n",
       "14               polyA RNA                         10xV3        oligo-dT   \n",
       "15               polyA RNA                         10xV3        oligo-dT   \n",
       "16               polyA RNA                         10xV3        oligo-dT   \n",
       "17               polyA RNA                         10xV3        oligo-dT   \n",
       "\n",
       "    Comment[sample barcode offset] Comment[sample barcode read]  \\\n",
       "0                                0                      index 1   \n",
       "1                                0                      index 1   \n",
       "2                                0                      index 1   \n",
       "3                                0                      index 1   \n",
       "4                                0                      index 1   \n",
       "5                                0                      index 1   \n",
       "6                                0                      index 1   \n",
       "7                                0                      index 1   \n",
       "8                                0                      index 1   \n",
       "9                                0                      index 1   \n",
       "10                               0                      index 1   \n",
       "11                               0                      index 1   \n",
       "12                               0                      index 1   \n",
       "13                               0                      index 1   \n",
       "14                               0                      index 1   \n",
       "15                               0                      index 1   \n",
       "16                               0                      index 1   \n",
       "17                               0                      index 1   \n",
       "\n",
       "    Comment[sample barcode size] Comment[single cell isolation]  \\\n",
       "0                              8                            10x   \n",
       "1                              8                            10x   \n",
       "2                              8                            10x   \n",
       "3                              8                            10x   \n",
       "4                              8                            10x   \n",
       "5                              8                            10x   \n",
       "6                              8                            10x   \n",
       "7                              8                            10x   \n",
       "8                              8                            10x   \n",
       "9                              8                            10x   \n",
       "10                             8                            10x   \n",
       "11                             8                            10x   \n",
       "12                             8                            10x   \n",
       "13                             8                            10x   \n",
       "14                             8                            10x   \n",
       "15                             8                            10x   \n",
       "16                             8                            10x   \n",
       "17                             8                            10x   \n",
       "\n",
       "   Comment[spike in]  Comment[umi barcode offset] Comment[umi barcode read]  \\\n",
       "0               none                           16                    read 1   \n",
       "1               none                           16                    read 1   \n",
       "2               none                           16                    read 1   \n",
       "3               none                           16                    read 1   \n",
       "4               none                           16                    read 1   \n",
       "5               none                           16                    read 1   \n",
       "6               none                           16                    read 1   \n",
       "7               none                           16                    read 1   \n",
       "8               none                           16                    read 1   \n",
       "9               none                           16                    read 1   \n",
       "10              none                           16                    read 1   \n",
       "11              none                           16                    read 1   \n",
       "12              none                           16                    read 1   \n",
       "13              none                           16                    read 1   \n",
       "14              none                           16                    read 1   \n",
       "15              none                           16                    read 1   \n",
       "16              none                           16                    read 1   \n",
       "17              none                           16                    read 1   \n",
       "\n",
       "    Comment[umi barcode size] Protocol REF.3                Performer.3  \\\n",
       "0                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "1                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "2                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "3                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "4                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "5                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "6                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "7                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "8                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "9                          12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "10                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "11                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "12                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "13                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "14                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "15                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "16                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "17                         12  P-MTAB-115983  Wellcome Sanger Institute   \n",
       "\n",
       "         Assay Name   Technology Type Comment[ENA_EXPERIMENT]  \\\n",
       "0   5705STDY8058280  sequencing assay              ERX6821446   \n",
       "1   5705STDY8058280  sequencing assay              ERX6821446   \n",
       "2   5705STDY8058280  sequencing assay              ERX6821446   \n",
       "3   5705STDY8058281  sequencing assay              ERX6821447   \n",
       "4   5705STDY8058281  sequencing assay              ERX6821447   \n",
       "5   5705STDY8058281  sequencing assay              ERX6821447   \n",
       "6   5705STDY8058282  sequencing assay              ERX6821448   \n",
       "7   5705STDY8058282  sequencing assay              ERX6821448   \n",
       "8   5705STDY8058282  sequencing assay              ERX6821448   \n",
       "9   5705STDY8058283  sequencing assay              ERX6821449   \n",
       "10  5705STDY8058283  sequencing assay              ERX6821449   \n",
       "11  5705STDY8058283  sequencing assay              ERX6821449   \n",
       "12  5705STDY8058284  sequencing assay              ERX6821450   \n",
       "13  5705STDY8058284  sequencing assay              ERX6821450   \n",
       "14  5705STDY8058284  sequencing assay              ERX6821450   \n",
       "15  5705STDY8058285  sequencing assay              ERX6821451   \n",
       "16  5705STDY8058285  sequencing assay              ERX6821451   \n",
       "17  5705STDY8058285  sequencing assay              ERX6821451   \n",
       "\n",
       "                      Scan Name Comment[SUBMITTED_FILE_NAME] Comment[ENA_RUN]  \\\n",
       "0   5705STDY8058280_I1.fastq.gz  5705STDY8058280_I1.fastq.gz       ERR7251990   \n",
       "1   5705STDY8058280_R1.fastq.gz  5705STDY8058280_R1.fastq.gz       ERR7251990   \n",
       "2   5705STDY8058280_R2.fastq.gz  5705STDY8058280_R2.fastq.gz       ERR7251990   \n",
       "3   5705STDY8058281_I1.fastq.gz  5705STDY8058281_I1.fastq.gz       ERR7251991   \n",
       "4   5705STDY8058281_R1.fastq.gz  5705STDY8058281_R1.fastq.gz       ERR7251991   \n",
       "5   5705STDY8058281_R2.fastq.gz  5705STDY8058281_R2.fastq.gz       ERR7251991   \n",
       "6   5705STDY8058282_I1.fastq.gz  5705STDY8058282_I1.fastq.gz       ERR7251992   \n",
       "7   5705STDY8058282_R1.fastq.gz  5705STDY8058282_R1.fastq.gz       ERR7251992   \n",
       "8   5705STDY8058282_R2.fastq.gz  5705STDY8058282_R2.fastq.gz       ERR7251992   \n",
       "9   5705STDY8058283_I1.fastq.gz  5705STDY8058283_I1.fastq.gz       ERR7251993   \n",
       "10  5705STDY8058283_R1.fastq.gz  5705STDY8058283_R1.fastq.gz       ERR7251993   \n",
       "11  5705STDY8058283_R2.fastq.gz  5705STDY8058283_R2.fastq.gz       ERR7251993   \n",
       "12  5705STDY8058284_I1.fastq.gz  5705STDY8058284_I1.fastq.gz       ERR7251994   \n",
       "13  5705STDY8058284_R1.fastq.gz  5705STDY8058284_R1.fastq.gz       ERR7251994   \n",
       "14  5705STDY8058284_R2.fastq.gz  5705STDY8058284_R2.fastq.gz       ERR7251994   \n",
       "15  5705STDY8058285_I1.fastq.gz  5705STDY8058285_I1.fastq.gz       ERR7251995   \n",
       "16  5705STDY8058285_R1.fastq.gz  5705STDY8058285_R1.fastq.gz       ERR7251995   \n",
       "17  5705STDY8058285_R2.fastq.gz  5705STDY8058285_R2.fastq.gz       ERR7251995   \n",
       "\n",
       "                                   Comment[FASTQ_URI]  \\\n",
       "0   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "1   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "2   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "3   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "4   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "5   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "6   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "7   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "8   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "9   ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "10  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "11  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "12  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "13  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "14  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "15  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "16  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "17  ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR725/ERR725...   \n",
       "\n",
       "            Comment[read1 file]          Comment[read2 file]  \\\n",
       "0   5705STDY8058280_R1.fastq.gz  5705STDY8058280_R2.fastq.gz   \n",
       "1   5705STDY8058280_R1.fastq.gz  5705STDY8058280_R2.fastq.gz   \n",
       "2   5705STDY8058280_R1.fastq.gz  5705STDY8058280_R2.fastq.gz   \n",
       "3   5705STDY8058281_R1.fastq.gz  5705STDY8058281_R2.fastq.gz   \n",
       "4   5705STDY8058281_R1.fastq.gz  5705STDY8058281_R2.fastq.gz   \n",
       "5   5705STDY8058281_R1.fastq.gz  5705STDY8058281_R2.fastq.gz   \n",
       "6   5705STDY8058282_R1.fastq.gz  5705STDY8058282_R2.fastq.gz   \n",
       "7   5705STDY8058282_R1.fastq.gz  5705STDY8058282_R2.fastq.gz   \n",
       "8   5705STDY8058282_R1.fastq.gz  5705STDY8058282_R2.fastq.gz   \n",
       "9   5705STDY8058283_R1.fastq.gz  5705STDY8058283_R2.fastq.gz   \n",
       "10  5705STDY8058283_R1.fastq.gz  5705STDY8058283_R2.fastq.gz   \n",
       "11  5705STDY8058283_R1.fastq.gz  5705STDY8058283_R2.fastq.gz   \n",
       "12  5705STDY8058284_R1.fastq.gz  5705STDY8058284_R2.fastq.gz   \n",
       "13  5705STDY8058284_R1.fastq.gz  5705STDY8058284_R2.fastq.gz   \n",
       "14  5705STDY8058284_R1.fastq.gz  5705STDY8058284_R2.fastq.gz   \n",
       "15  5705STDY8058285_R1.fastq.gz  5705STDY8058285_R2.fastq.gz   \n",
       "16  5705STDY8058285_R1.fastq.gz  5705STDY8058285_R2.fastq.gz   \n",
       "17  5705STDY8058285_R1.fastq.gz  5705STDY8058285_R2.fastq.gz   \n",
       "\n",
       "           Comment[index1 file] Protocol REF.4                Performer.4  \\\n",
       "0   5705STDY8058280_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "1   5705STDY8058280_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "2   5705STDY8058280_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "3   5705STDY8058281_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "4   5705STDY8058281_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "5   5705STDY8058281_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "6   5705STDY8058282_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "7   5705STDY8058282_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "8   5705STDY8058282_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "9   5705STDY8058283_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "10  5705STDY8058283_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "11  5705STDY8058283_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "12  5705STDY8058284_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "13  5705STDY8058284_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "14  5705STDY8058284_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "15  5705STDY8058285_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "16  5705STDY8058285_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "17  5705STDY8058285_I1.fastq.gz  P-MTAB-115984  Wellcome Sanger Institute   \n",
       "\n",
       "   Protocol REF.5                        Derived Array Data File  \\\n",
       "0   P-MTAB-115985  5705STDY8058280_filtered_feature_bc_matrix.h5   \n",
       "1   P-MTAB-115985  5705STDY8058280_filtered_feature_bc_matrix.h5   \n",
       "2   P-MTAB-115985  5705STDY8058280_filtered_feature_bc_matrix.h5   \n",
       "3   P-MTAB-115985  5705STDY8058281_filtered_feature_bc_matrix.h5   \n",
       "4   P-MTAB-115985  5705STDY8058281_filtered_feature_bc_matrix.h5   \n",
       "5   P-MTAB-115985  5705STDY8058281_filtered_feature_bc_matrix.h5   \n",
       "6   P-MTAB-115985  5705STDY8058282_filtered_feature_bc_matrix.h5   \n",
       "7   P-MTAB-115985  5705STDY8058282_filtered_feature_bc_matrix.h5   \n",
       "8   P-MTAB-115985  5705STDY8058282_filtered_feature_bc_matrix.h5   \n",
       "9   P-MTAB-115985  5705STDY8058283_filtered_feature_bc_matrix.h5   \n",
       "10  P-MTAB-115985  5705STDY8058283_filtered_feature_bc_matrix.h5   \n",
       "11  P-MTAB-115985  5705STDY8058283_filtered_feature_bc_matrix.h5   \n",
       "12  P-MTAB-115985  5705STDY8058284_filtered_feature_bc_matrix.h5   \n",
       "13  P-MTAB-115985  5705STDY8058284_filtered_feature_bc_matrix.h5   \n",
       "14  P-MTAB-115985  5705STDY8058284_filtered_feature_bc_matrix.h5   \n",
       "15  P-MTAB-115985  5705STDY8058285_filtered_feature_bc_matrix.h5   \n",
       "16  P-MTAB-115985  5705STDY8058285_filtered_feature_bc_matrix.h5   \n",
       "17  P-MTAB-115985  5705STDY8058285_filtered_feature_bc_matrix.h5   \n",
       "\n",
       "              Comment [Derived ArrayExpress FTP file] Protocol REF.6  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "15  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "16  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "17  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "\n",
       "                  Performer.5 Protocol REF.7  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "15  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "16  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "17  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "\n",
       "           Derived Array Data File.1  \\\n",
       "0   5705STDY8058280_molecule_info.h5   \n",
       "1   5705STDY8058280_molecule_info.h5   \n",
       "2   5705STDY8058280_molecule_info.h5   \n",
       "3   5705STDY8058281_molecule_info.h5   \n",
       "4   5705STDY8058281_molecule_info.h5   \n",
       "5   5705STDY8058281_molecule_info.h5   \n",
       "6   5705STDY8058282_molecule_info.h5   \n",
       "7   5705STDY8058282_molecule_info.h5   \n",
       "8   5705STDY8058282_molecule_info.h5   \n",
       "9   5705STDY8058283_molecule_info.h5   \n",
       "10  5705STDY8058283_molecule_info.h5   \n",
       "11  5705STDY8058283_molecule_info.h5   \n",
       "12  5705STDY8058284_molecule_info.h5   \n",
       "13  5705STDY8058284_molecule_info.h5   \n",
       "14  5705STDY8058284_molecule_info.h5   \n",
       "15  5705STDY8058285_molecule_info.h5   \n",
       "16  5705STDY8058285_molecule_info.h5   \n",
       "17  5705STDY8058285_molecule_info.h5   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].1 Protocol REF.8  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "15  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "16  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "17  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...  P-MTAB-115984   \n",
       "\n",
       "                  Performer.6 Protocol REF.9  \\\n",
       "0   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "1   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "2   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "3   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "4   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "5   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "6   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "7   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "8   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "9   Wellcome Sanger Institute  P-MTAB-115985   \n",
       "10  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "11  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "12  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "13  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "14  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "15  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "16  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "17  Wellcome Sanger Institute  P-MTAB-115985   \n",
       "\n",
       "              Derived Array Data File.2  \\\n",
       "0   5705STDY8058280_metrics_summary.csv   \n",
       "1   5705STDY8058280_metrics_summary.csv   \n",
       "2   5705STDY8058280_metrics_summary.csv   \n",
       "3   5705STDY8058281_metrics_summary.csv   \n",
       "4   5705STDY8058281_metrics_summary.csv   \n",
       "5   5705STDY8058281_metrics_summary.csv   \n",
       "6   5705STDY8058282_metrics_summary.csv   \n",
       "7   5705STDY8058282_metrics_summary.csv   \n",
       "8   5705STDY8058282_metrics_summary.csv   \n",
       "9   5705STDY8058283_metrics_summary.csv   \n",
       "10  5705STDY8058283_metrics_summary.csv   \n",
       "11  5705STDY8058283_metrics_summary.csv   \n",
       "12  5705STDY8058284_metrics_summary.csv   \n",
       "13  5705STDY8058284_metrics_summary.csv   \n",
       "14  5705STDY8058284_metrics_summary.csv   \n",
       "15  5705STDY8058285_metrics_summary.csv   \n",
       "16  5705STDY8058285_metrics_summary.csv   \n",
       "17  5705STDY8058285_metrics_summary.csv   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].2 Protocol REF.10  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "15  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "16  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "17  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "\n",
       "                  Performer.7 Protocol REF.11  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "15  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "16  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "17  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "\n",
       "                   Derived Array Data File.3  \\\n",
       "0   5705STDY8058280_raw_feature_bc_matrix.h5   \n",
       "1   5705STDY8058280_raw_feature_bc_matrix.h5   \n",
       "2   5705STDY8058280_raw_feature_bc_matrix.h5   \n",
       "3   5705STDY8058281_raw_feature_bc_matrix.h5   \n",
       "4   5705STDY8058281_raw_feature_bc_matrix.h5   \n",
       "5   5705STDY8058281_raw_feature_bc_matrix.h5   \n",
       "6   5705STDY8058282_raw_feature_bc_matrix.h5   \n",
       "7   5705STDY8058282_raw_feature_bc_matrix.h5   \n",
       "8   5705STDY8058282_raw_feature_bc_matrix.h5   \n",
       "9   5705STDY8058283_raw_feature_bc_matrix.h5   \n",
       "10  5705STDY8058283_raw_feature_bc_matrix.h5   \n",
       "11  5705STDY8058283_raw_feature_bc_matrix.h5   \n",
       "12  5705STDY8058284_raw_feature_bc_matrix.h5   \n",
       "13  5705STDY8058284_raw_feature_bc_matrix.h5   \n",
       "14  5705STDY8058284_raw_feature_bc_matrix.h5   \n",
       "15  5705STDY8058285_raw_feature_bc_matrix.h5   \n",
       "16  5705STDY8058285_raw_feature_bc_matrix.h5   \n",
       "17  5705STDY8058285_raw_feature_bc_matrix.h5   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].3 Protocol REF.12  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "15  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "16  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "17  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "\n",
       "                  Performer.8 Protocol REF.13  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115985   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "15  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "16  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "17  Wellcome Sanger Institute   P-MTAB-115985   \n",
       "\n",
       "           Derived Array Data File.4  \\\n",
       "0   5705STDY8058280_web_summary.html   \n",
       "1   5705STDY8058280_web_summary.html   \n",
       "2   5705STDY8058280_web_summary.html   \n",
       "3   5705STDY8058281_web_summary.html   \n",
       "4   5705STDY8058281_web_summary.html   \n",
       "5   5705STDY8058281_web_summary.html   \n",
       "6   5705STDY8058282_web_summary.html   \n",
       "7   5705STDY8058282_web_summary.html   \n",
       "8   5705STDY8058282_web_summary.html   \n",
       "9   5705STDY8058283_web_summary.html   \n",
       "10  5705STDY8058283_web_summary.html   \n",
       "11  5705STDY8058283_web_summary.html   \n",
       "12  5705STDY8058284_web_summary.html   \n",
       "13  5705STDY8058284_web_summary.html   \n",
       "14  5705STDY8058284_web_summary.html   \n",
       "15  5705STDY8058285_web_summary.html   \n",
       "16  5705STDY8058285_web_summary.html   \n",
       "17  5705STDY8058285_web_summary.html   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].4 Protocol REF.14  \\\n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "15  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "16  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "17  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   P-MTAB-115984   \n",
       "\n",
       "                  Performer.9 Protocol REF.15 Derived Array Data File.5  \\\n",
       "0   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "1   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "2   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "3   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "4   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "5   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "6   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "7   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "8   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "9   Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "10  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "11  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "12  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "13  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "14  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "15  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "16  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "17  Wellcome Sanger Institute   P-MTAB-115985       cell_annotation.csv   \n",
       "\n",
       "            Comment [Derived ArrayExpress FTP file].5 Factor Value[individual]  \n",
       "0   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "1   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "2   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "3   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "4   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "5   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "6   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "7   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "8   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 1  \n",
       "9   ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "10  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "11  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "12  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "13  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "14  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "15  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "16  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  \n",
       "17  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...             Individual 2  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SN_meta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "715772e2",
   "metadata": {},
   "source": [
    "# ST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "da2e06cf",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before: (2987, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/preprocessing/_normalization.py:170: UserWarning: Received a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After: (2983, 19375)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before: (3499, 31053)\n",
      "After: (3495, 19268)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before: (3497, 31053)\n",
      "After: (3492, 19587)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before: (2409, 31053)\n",
      "After: (2409, 18966)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before: (2576, 31053)\n",
      "After: (2575, 19306)\n"
     ]
    }
   ],
   "source": [
    "fs = glob('../Dataset/C2L/ST/*/*h5')\n",
    "\n",
    "STs = []\n",
    "\n",
    "for f in fs:\n",
    "    f = f.split('/')\n",
    "\n",
    "    ST = sc.read_visium(f[0] + '/' + f[1], count_file = f[2], load_images = True)\n",
    "    print('Before:', ST.shape)\n",
    "    \n",
    "    ST.var_names = ST.var_names.str.upper()\n",
    "    ST.var_names_make_unique()\n",
    "    sc.pp.filter_cells(ST, min_genes=200)\n",
    "    sc.pp.filter_genes(ST, min_cells=3)\n",
    "    ST.var['mt'] = ST.var_names.str.startswith('mt-')\n",
    "    sc.pp.calculate_qc_metrics(ST, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)\n",
    "    \n",
    "    ST = ST[ST.obs.pct_counts_mt < 25, :]\n",
    "    ST = ST[:, ~ST.var['mt']]\n",
    "    \n",
    "    sc.pp.normalize_total(ST, target_sum=1e4)\n",
    "    sc.pp.log1p(ST)\n",
    "    \n",
    "    #sc.pp.highly_variable_genes(ST, n_top_genes = 6000)\n",
    "    #ST = ST[:, ST.var.highly_variable]\n",
    "    \n",
    "    print('After:', ST.shape)\n",
    "\n",
    "    STs.append(ST)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89b2c2a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "rcParams[\"axes.facecolor\"] = \"black\"\n",
    "rcParams[\"figure.figsize\"] = [6,7]\n",
    "rcParams[\"font.size\"] = 28\n",
    "\n",
    "for ST in STs:\n",
    "    sc.pl.spatial(ST, \n",
    "                  color=[\"RORB\"], img_key='hires',\n",
    "                  vmin=0, cmap='magma', vmax='p99.0',\n",
    "                  gene_symbols='SYMBOL', size=0.8, colorbar_loc = None\n",
    "                 )\n",
    "    \n",
    "rcParams[\"axes.facecolor\"] = \"white\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21584a48",
   "metadata": {},
   "source": [
    "# SN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4685d927",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(8809, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(8913, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6233, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5751, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3284, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11439, 31053)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/anndata/_core/anndata.py:801: UserWarning: \n",
      "AnnData expects .obs.index to contain strings, but got values like:\n",
      "    [0, 1, 2, 3, 4]\n",
      "\n",
      "    Inferred to be: integer\n",
      "\n",
      "  value_idx = self._prep_dim_index(value.index, attr)\n"
     ]
    }
   ],
   "source": [
    "fs = glob('../Dataset/C2L/SN/5/*h5')\n",
    "\n",
    "SNs = []\n",
    "for f in fs:\n",
    "    SN = sc.read_10x_h5(f)\n",
    "    SN.var_names = SN.var_names.str.upper()\n",
    "    SN.var_names_make_unique()\n",
    "    \n",
    "    SN.obs['Cell ID'] = f.split('/')[-1].split('_')[0] + '_' + SN.obs.index\n",
    "    SN.obs.index = f.split('/')[-1].split('_')[0] + '_' + SN.obs.index\n",
    "    \n",
    "    print(SN.shape)\n",
    "\n",
    "    SNs.append(SN)\n",
    "    \n",
    "SN = SNs[len(fs)-1]\n",
    "for i in range(len(fs)-1):\n",
    "    SN = SN.concatenate(SNs[i])\n",
    "    \n",
    "Anno = pd.read_csv('../Dataset/C2L/SN/4/cell_annotation.csv')\n",
    "index = SN.obs.index\n",
    "SN.obs = SN.obs.merge(Anno, how = 'left')\n",
    "SN.obs.index = index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "861b8f16",
   "metadata": {},
   "outputs": [],
   "source": [
    "print('Before:', SN.shape)\n",
    "\n",
    "sc.pp.filter_cells(SN, min_genes=200)\n",
    "sc.pp.filter_genes(SN, min_cells=3)\n",
    "SN.var['mt'] = SN.var_names.str.startswith('mt-')\n",
    "sc.pp.calculate_qc_metrics(SN, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)\n",
    "\n",
    "sc.pl.violin(SN, ['n_genes_by_counts', 'total_counts', 'pct_counts_mt'],\n",
    "             jitter=0.4, multi_panel=True)\n",
    "\n",
    "sc.pl.scatter(SN, x='total_counts', y='pct_counts_mt')\n",
    "sc.pl.scatter(SN, x='total_counts', y='n_genes_by_counts')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3850fcd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "SN = SN[SN.obs.n_genes_by_counts < 6500, :]\n",
    "SN = SN[SN.obs.pct_counts_mt < 5, :]\n",
    "\n",
    "SN = SN[:, ~SN.var['mt']]\n",
    "\n",
    "sc.pp.normalize_total(SN, target_sum=1e4)\n",
    "sc.pp.log1p(SN)\n",
    "\n",
    "#sc.pp.highly_variable_genes(SN, n_top_genes = 6000)\n",
    "#SN = SN[:, SN.var.highly_variable]\n",
    "\n",
    "print('After:', SN.shape)\n",
    "sc.pl.scatter(SN, x='total_counts', y='pct_counts_mt')\n",
    "sc.pl.scatter(SN, x='total_counts', y='n_genes_by_counts')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "446d6655",
   "metadata": {},
   "outputs": [],
   "source": [
    "sc.tl.pca(SN, svd_solver='arpack')\n",
    "sc.pl.pca(SN, color='annotation_1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "830e3f6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "sc.pp.neighbors(SN, n_neighbors=10, n_pcs=40)\n",
    "\n",
    "sc.tl.umap(SN)\n",
    "sc.pl.umap(SN, color=['annotation_1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7b21ea1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "SN = SN[~SN.obs['sample'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "151f7312",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell ID</th>\n",
       "      <th>batch</th>\n",
       "      <th>sample</th>\n",
       "      <th>annotation_1</th>\n",
       "      <th>annotation_1_print</th>\n",
       "      <th>n_genes</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>total_counts_mt</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0</th>\n",
       "      <td>5705STDY8058285_AAACCCAAGACTCATC-1</td>\n",
       "      <td>0</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>Ext_Unk_3</td>\n",
       "      <td>36_Ext_Unk_3</td>\n",
       "      <td>2155</td>\n",
       "      <td>2155</td>\n",
       "      <td>4085.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0</th>\n",
       "      <td>5705STDY8058285_AAACCCAAGCACTCTA-1</td>\n",
       "      <td>0</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>Ext_ClauPyr</td>\n",
       "      <td>16_Ext_ClauPyr</td>\n",
       "      <td>2151</td>\n",
       "      <td>2151</td>\n",
       "      <td>4853.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0</th>\n",
       "      <td>5705STDY8058285_AAACCCAAGGTGGGTT-1</td>\n",
       "      <td>0</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>Inh_3</td>\n",
       "      <td>39_Inh_3</td>\n",
       "      <td>1988</td>\n",
       "      <td>1988</td>\n",
       "      <td>3334.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0</th>\n",
       "      <td>5705STDY8058285_AAACCCAAGTTGTACC-1</td>\n",
       "      <td>0</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>Oligo_2</td>\n",
       "      <td>59_Oligo_2</td>\n",
       "      <td>1217</td>\n",
       "      <td>1217</td>\n",
       "      <td>2478.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0</th>\n",
       "      <td>5705STDY8058285_AAACCCACAACTCGAT-1</td>\n",
       "      <td>0</td>\n",
       "      <td>5705STDY8058285</td>\n",
       "      <td>Ext_Amy_1</td>\n",
       "      <td>14_Ext_Amy_1</td>\n",
       "      <td>4642</td>\n",
       "      <td>4641</td>\n",
       "      <td>13857.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058284_TTTGGTTGTTAAGGAT-1-1</th>\n",
       "      <td>5705STDY8058284_TTTGGTTGTTAAGGAT-1</td>\n",
       "      <td>1</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>Oligo_2</td>\n",
       "      <td>59_Oligo_2</td>\n",
       "      <td>1400</td>\n",
       "      <td>1400</td>\n",
       "      <td>2821.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058284_TTTGTTGAGGGAGTTC-1-1</th>\n",
       "      <td>5705STDY8058284_TTTGTTGAGGGAGTTC-1</td>\n",
       "      <td>1</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>Oligo_2</td>\n",
       "      <td>59_Oligo_2</td>\n",
       "      <td>1418</td>\n",
       "      <td>1418</td>\n",
       "      <td>2602.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058284_TTTGTTGCATTCACCC-1-1</th>\n",
       "      <td>5705STDY8058284_TTTGTTGCATTCACCC-1</td>\n",
       "      <td>1</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>Ext_L25</td>\n",
       "      <td>23_Ext_L25</td>\n",
       "      <td>4084</td>\n",
       "      <td>4083</td>\n",
       "      <td>12459.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058284_TTTGTTGGTCTAGGTT-1-1</th>\n",
       "      <td>5705STDY8058284_TTTGTTGGTCTAGGTT-1</td>\n",
       "      <td>1</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>Ext_L6</td>\n",
       "      <td>28_Ext_L6</td>\n",
       "      <td>3804</td>\n",
       "      <td>3804</td>\n",
       "      <td>11445.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5705STDY8058284_TTTGTTGGTTCAAGTC-1-1</th>\n",
       "      <td>5705STDY8058284_TTTGTTGGTTCAAGTC-1</td>\n",
       "      <td>1</td>\n",
       "      <td>5705STDY8058284</td>\n",
       "      <td>Ext_Thal_2</td>\n",
       "      <td>33_Ext_Thal_2</td>\n",
       "      <td>4407</td>\n",
       "      <td>4407</td>\n",
       "      <td>13952.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>40532 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         Cell ID  \\\n",
       "5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0  5705STDY8058285_AAACCCAAGACTCATC-1   \n",
       "5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0  5705STDY8058285_AAACCCAAGCACTCTA-1   \n",
       "5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0  5705STDY8058285_AAACCCAAGGTGGGTT-1   \n",
       "5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0  5705STDY8058285_AAACCCAAGTTGTACC-1   \n",
       "5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0  5705STDY8058285_AAACCCACAACTCGAT-1   \n",
       "...                                                                          ...   \n",
       "5705STDY8058284_TTTGGTTGTTAAGGAT-1-1          5705STDY8058284_TTTGGTTGTTAAGGAT-1   \n",
       "5705STDY8058284_TTTGTTGAGGGAGTTC-1-1          5705STDY8058284_TTTGTTGAGGGAGTTC-1   \n",
       "5705STDY8058284_TTTGTTGCATTCACCC-1-1          5705STDY8058284_TTTGTTGCATTCACCC-1   \n",
       "5705STDY8058284_TTTGTTGGTCTAGGTT-1-1          5705STDY8058284_TTTGTTGGTCTAGGTT-1   \n",
       "5705STDY8058284_TTTGTTGGTTCAAGTC-1-1          5705STDY8058284_TTTGTTGGTTCAAGTC-1   \n",
       "\n",
       "                                             batch           sample  \\\n",
       "5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0     0  5705STDY8058285   \n",
       "5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0     0  5705STDY8058285   \n",
       "5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0     0  5705STDY8058285   \n",
       "5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0     0  5705STDY8058285   \n",
       "5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0     0  5705STDY8058285   \n",
       "...                                            ...              ...   \n",
       "5705STDY8058284_TTTGGTTGTTAAGGAT-1-1             1  5705STDY8058284   \n",
       "5705STDY8058284_TTTGTTGAGGGAGTTC-1-1             1  5705STDY8058284   \n",
       "5705STDY8058284_TTTGTTGCATTCACCC-1-1             1  5705STDY8058284   \n",
       "5705STDY8058284_TTTGTTGGTCTAGGTT-1-1             1  5705STDY8058284   \n",
       "5705STDY8058284_TTTGTTGGTTCAAGTC-1-1             1  5705STDY8058284   \n",
       "\n",
       "                                             annotation_1 annotation_1_print  \\\n",
       "5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0    Ext_Unk_3       36_Ext_Unk_3   \n",
       "5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0  Ext_ClauPyr     16_Ext_ClauPyr   \n",
       "5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0        Inh_3           39_Inh_3   \n",
       "5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0      Oligo_2         59_Oligo_2   \n",
       "5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0    Ext_Amy_1       14_Ext_Amy_1   \n",
       "...                                                   ...                ...   \n",
       "5705STDY8058284_TTTGGTTGTTAAGGAT-1-1              Oligo_2         59_Oligo_2   \n",
       "5705STDY8058284_TTTGTTGAGGGAGTTC-1-1              Oligo_2         59_Oligo_2   \n",
       "5705STDY8058284_TTTGTTGCATTCACCC-1-1              Ext_L25         23_Ext_L25   \n",
       "5705STDY8058284_TTTGTTGGTCTAGGTT-1-1               Ext_L6          28_Ext_L6   \n",
       "5705STDY8058284_TTTGTTGGTTCAAGTC-1-1           Ext_Thal_2      33_Ext_Thal_2   \n",
       "\n",
       "                                              n_genes  n_genes_by_counts  \\\n",
       "5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0     2155               2155   \n",
       "5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0     2151               2151   \n",
       "5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0     1988               1988   \n",
       "5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0     1217               1217   \n",
       "5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0     4642               4641   \n",
       "...                                               ...                ...   \n",
       "5705STDY8058284_TTTGGTTGTTAAGGAT-1-1             1400               1400   \n",
       "5705STDY8058284_TTTGTTGAGGGAGTTC-1-1             1418               1418   \n",
       "5705STDY8058284_TTTGTTGCATTCACCC-1-1             4084               4083   \n",
       "5705STDY8058284_TTTGTTGGTCTAGGTT-1-1             3804               3804   \n",
       "5705STDY8058284_TTTGTTGGTTCAAGTC-1-1             4407               4407   \n",
       "\n",
       "                                              total_counts  total_counts_mt  \\\n",
       "5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0        4085.0              0.0   \n",
       "5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0        4853.0              0.0   \n",
       "5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0        3334.0              0.0   \n",
       "5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0        2478.0              0.0   \n",
       "5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0       13857.0              0.0   \n",
       "...                                                    ...              ...   \n",
       "5705STDY8058284_TTTGGTTGTTAAGGAT-1-1                2821.0              0.0   \n",
       "5705STDY8058284_TTTGTTGAGGGAGTTC-1-1                2602.0              0.0   \n",
       "5705STDY8058284_TTTGTTGCATTCACCC-1-1               12459.0              0.0   \n",
       "5705STDY8058284_TTTGTTGGTCTAGGTT-1-1               11445.0              0.0   \n",
       "5705STDY8058284_TTTGTTGGTTCAAGTC-1-1               13952.0              0.0   \n",
       "\n",
       "                                              pct_counts_mt  \n",
       "5705STDY8058285_AAACCCAAGACTCATC-1-0-0-0-0-0            0.0  \n",
       "5705STDY8058285_AAACCCAAGCACTCTA-1-0-0-0-0-0            0.0  \n",
       "5705STDY8058285_AAACCCAAGGTGGGTT-1-0-0-0-0-0            0.0  \n",
       "5705STDY8058285_AAACCCAAGTTGTACC-1-0-0-0-0-0            0.0  \n",
       "5705STDY8058285_AAACCCACAACTCGAT-1-0-0-0-0-0            0.0  \n",
       "...                                                     ...  \n",
       "5705STDY8058284_TTTGGTTGTTAAGGAT-1-1                    0.0  \n",
       "5705STDY8058284_TTTGTTGAGGGAGTTC-1-1                    0.0  \n",
       "5705STDY8058284_TTTGTTGCATTCACCC-1-1                    0.0  \n",
       "5705STDY8058284_TTTGTTGGTCTAGGTT-1-1                    0.0  \n",
       "5705STDY8058284_TTTGTTGGTTCAAGTC-1-1                    0.0  \n",
       "\n",
       "[40532 rows x 10 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SN.obs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d267c12",
   "metadata": {},
   "source": [
    "# Check Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "9820d013",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function spatial in module scanpy.plotting._tools.scatterplots:\n",
      "\n",
      "spatial(adata: anndata._core.anndata.AnnData, *, color: Union[str, Sequence[str], NoneType] = None, gene_symbols: Optional[str] = None, use_raw: Optional[bool] = None, sort_order: bool = True, edges: bool = False, edges_width: float = 0.1, edges_color: Union[str, Sequence[float], Sequence[str]] = 'grey', neighbors_key: Optional[str] = None, arrows: bool = False, arrows_kwds: Optional[Mapping[str, Any]] = None, groups: Optional[str] = None, components: Union[str, Sequence[str]] = None, dimensions: Union[Tuple[int, int], Sequence[Tuple[int, int]], NoneType] = None, layer: Optional[str] = None, projection: Literal['2d', '3d'] = '2d', scale_factor: Optional[float] = None, color_map: Union[matplotlib.colors.Colormap, str, NoneType] = None, cmap: Union[matplotlib.colors.Colormap, str, NoneType] = None, palette: Union[str, Sequence[str], cycler.Cycler, NoneType] = None, na_color: Union[str, Tuple[float, ...], NoneType] = None, na_in_legend: bool = True, size: float = 1.0, frameon: Optional[bool] = None, legend_fontsize: Union[int, float, Literal['xx-small', 'x-small', 'small', 'medium', 'large', 'x-large', 'xx-large'], NoneType] = None, legend_fontweight: Union[int, Literal['light', 'normal', 'medium', 'semibold', 'bold', 'heavy', 'black']] = 'bold', legend_loc: str = 'right margin', legend_fontoutline: Optional[int] = None, colorbar_loc: Optional[str] = 'right', vmax: Union[str, float, Callable[[Sequence[float]], float], Sequence[Union[str, float, Callable[[Sequence[float]], float]]], NoneType] = None, vmin: Union[str, float, Callable[[Sequence[float]], float], Sequence[Union[str, float, Callable[[Sequence[float]], float]]], NoneType] = None, vcenter: Union[str, float, Callable[[Sequence[float]], float], Sequence[Union[str, float, Callable[[Sequence[float]], float]]], NoneType] = None, norm: Union[matplotlib.colors.Normalize, Sequence[matplotlib.colors.Normalize], NoneType] = None, add_outline: Optional[bool] = False, outline_width: Tuple[float, float] = (0.3, 0.05), outline_color: Tuple[str, str] = ('black', 'white'), ncols: int = 4, hspace: float = 0.25, wspace: Optional[float] = None, title: Union[str, Sequence[str], NoneType] = None, show: Optional[bool] = None, save: Union[bool, str, NoneType] = None, ax: Optional[matplotlib.axes._axes.Axes] = None, return_fig: Optional[bool] = None, basis: str = 'spatial', img: Optional[numpy.ndarray] = None, img_key: Union[str, NoneType, scanpy._utils.Empty] = <Empty.token: 0>, library_id: Union[str, scanpy._utils.Empty] = <Empty.token: 0>, crop_coord: Tuple[int, int, int, int] = None, alpha_img: float = 1.0, bw: Optional[bool] = False, spot_size: Optional[float] = None, **kwargs) -> Union[matplotlib.axes._axes.Axes, List[matplotlib.axes._axes.Axes], NoneType]\n",
      "    Scatter plot in spatial coordinates.\n",
      "    \n",
      "    This function allows overlaying data on top of images.\n",
      "    Use the parameter `img_key` to see the image in the background\n",
      "    And the parameter `library_id` to select the image.\n",
      "    By default, `'hires'` and `'lowres'` are attempted.\n",
      "    \n",
      "    Use `crop_coord`, `alpha_img`, and `bw` to control how it is displayed.\n",
      "    Use `size` to scale the size of the Visium spots plotted on top.\n",
      "    \n",
      "    As this function is designed to for imaging data, there are two key assumptions\n",
      "    about how coordinates are handled:\n",
      "    \n",
      "    1. The origin (e.g `(0, 0)`) is at the top left  as is common convention\n",
      "    with image data.\n",
      "    \n",
      "    2. Coordinates are in the pixel space of the source image, so an equal\n",
      "    aspect ratio is assumed.\n",
      "    \n",
      "    If your anndata object has a `\"spatial\"` entry in `.uns`, the `img_key`\n",
      "    and `library_id` parameters to find values for `img`, `scale_factor`,\n",
      "    and `spot_size` arguments. Alternatively, these values be passed directly.\n",
      "    \n",
      "    Parameters\n",
      "    ----------\n",
      "    adata\n",
      "        Annotated data matrix.\n",
      "    color\n",
      "        Keys for annotations of observations/cells or variables/genes, e.g.,\n",
      "        `'ann1'` or `['ann1', 'ann2']`.\n",
      "    gene_symbols\n",
      "        Column name in `.var` DataFrame that stores gene symbols. By default `var_names`\n",
      "        refer to the index column of the `.var` DataFrame. Setting this option allows\n",
      "        alternative names to be used.\n",
      "    use_raw\n",
      "        Use `.raw` attribute of `adata` for coloring with gene expression. If `None`,\n",
      "        defaults to `True` if `layer` isn't provided and `adata.raw` is present.\n",
      "    layer\n",
      "        Name of the AnnData object layer that wants to be plotted. By default\n",
      "        adata.raw.X is plotted. If `use_raw=False` is set, then `adata.X` is plotted.\n",
      "        If `layer` is set to a valid layer name, then the layer is plotted. `layer`\n",
      "        takes precedence over `use_raw`.\n",
      "    library_id\n",
      "        library_id for Visium data, e.g. key in `adata.uns[\"spatial\"]`.\n",
      "    img_key\n",
      "        Key for image data, used to get `img` and `scale_factor` from `\"images\"`\n",
      "        and `\"scalefactors\"` entires for this library. To use spatial coordinates,\n",
      "        but not plot an image, pass `img_key=None`.\n",
      "    img\n",
      "        image data to plot, overrides `img_key`.\n",
      "    scale_factor\n",
      "        Scaling factor used to map from coordinate space to pixel space.\n",
      "        Found by default if `library_id` and `img_key` can be resolved.\n",
      "        Otherwise defaults to `1.`.\n",
      "    spot_size\n",
      "        Diameter of spot (in coordinate space) for each point. Diameter\n",
      "        in pixels of the spots will be `size * spot_size * scale_factor`.\n",
      "        This argument is required if it cannot be resolved from library info.\n",
      "    crop_coord\n",
      "        Coordinates to use for cropping the image (left, right, top, bottom).\n",
      "        These coordinates are expected to be in pixel space (same as `basis`)\n",
      "        and will be transformed by `scale_factor`.\n",
      "        If not provided, image is automatically cropped to bounds of `basis`,\n",
      "        plus a border.\n",
      "    alpha_img\n",
      "        Alpha value for image.\n",
      "    bw\n",
      "        Plot image data in gray scale.\n",
      "    sort_order\n",
      "        For continuous annotations used as color parameter, plot data points\n",
      "        with higher values on top of others.\n",
      "    groups\n",
      "        Restrict to a few categories in categorical observation annotation.\n",
      "        The default is not to restrict to any groups.\n",
      "    dimensions\n",
      "        0-indexed dimensions of the embedding to plot as integers. E.g. [(0, 1), (1, 2)].\n",
      "        Unlike `components`, this argument is used in the same way as `colors`, e.g. is\n",
      "        used to specify a single plot at a time. Will eventually replace the components\n",
      "        argument.\n",
      "    components\n",
      "        For instance, `['1,2', '2,3']`. To plot all available components use\n",
      "        `components='all'`.\n",
      "    projection\n",
      "        Projection of plot (default: `'2d'`).\n",
      "    legend_loc\n",
      "        Location of legend, either `'on data'`, `'right margin'` or a valid keyword\n",
      "        for the `loc` parameter of :class:`~matplotlib.legend.Legend`.\n",
      "    legend_fontsize\n",
      "        Numeric size in pt or string describing the size.\n",
      "        See :meth:`~matplotlib.text.Text.set_fontsize`.\n",
      "    legend_fontweight\n",
      "        Legend font weight. A numeric value in range 0-1000 or a string.\n",
      "        Defaults to `'bold'` if `legend_loc == 'on data'`, otherwise to `'normal'`.\n",
      "        See :meth:`~matplotlib.text.Text.set_fontweight`.\n",
      "    legend_fontoutline\n",
      "        Line width of the legend font outline in pt. Draws a white outline using\n",
      "        the path effect :class:`~matplotlib.patheffects.withStroke`.\n",
      "    colorbar_loc\n",
      "        Where to place the colorbar for continous variables. If `None`, no colorbar\n",
      "        is added.\n",
      "    size\n",
      "        Point size. If `None`, is automatically computed as 120000 / n_cells.\n",
      "        Can be a sequence containing the size for each cell. The order should be\n",
      "        the same as in adata.obs.\n",
      "    color_map\n",
      "        Color map to use for continous variables. Can be a name or a\n",
      "        :class:`~matplotlib.colors.Colormap` instance (e.g. `\"magma`\", `\"viridis\"`\n",
      "        or `mpl.cm.cividis`), see :func:`~matplotlib.cm.get_cmap`.\n",
      "        If `None`, the value of `mpl.rcParams[\"image.cmap\"]` is used.\n",
      "        The default `color_map` can be set using :func:`~scanpy.set_figure_params`.\n",
      "    palette\n",
      "        Colors to use for plotting categorical annotation groups.\n",
      "        The palette can be a valid :class:`~matplotlib.colors.ListedColormap` name\n",
      "        (`'Set2'`, `'tab20'`, ), a :class:`~cycler.Cycler` object, a dict mapping\n",
      "        categories to colors, or a sequence of colors. Colors must be valid to\n",
      "        matplotlib. (see :func:`~matplotlib.colors.is_color_like`).\n",
      "        If `None`, `mpl.rcParams[\"axes.prop_cycle\"]` is used unless the categorical\n",
      "        variable already has colors stored in `adata.uns[\"{var}_colors\"]`.\n",
      "        If provided, values of `adata.uns[\"{var}_colors\"]` will be set.\n",
      "    na_color\n",
      "        Color to use for null or masked values. Can be anything matplotlib accepts as a\n",
      "        color. Used for all points if `color=None`.\n",
      "    na_in_legend\n",
      "        If there are missing values, whether they get an entry in the legend. Currently\n",
      "        only implemented for categorical legends.\n",
      "    frameon\n",
      "        Draw a frame around the scatter plot. Defaults to value set in\n",
      "        :func:`~scanpy.set_figure_params`, defaults to `True`.\n",
      "    title\n",
      "        Provide title for panels either as string or list of strings,\n",
      "        e.g. `['title1', 'title2', ...]`.\n",
      "    \n",
      "    vmin\n",
      "        The value representing the lower limit of the color scale. Values smaller than vmin are plotted\n",
      "        with the same color as vmin. vmin can be a number, a string, a function or `None`. If\n",
      "        vmin is a string and has the format `pN`, this is interpreted as a vmin=percentile(N).\n",
      "        For example vmin='p1.5' is interpreted as the 1.5 percentile. If vmin is function, then\n",
      "        vmin is interpreted as the return value of the function over the list of values to plot.\n",
      "        For example to set vmin tp the mean of the values to plot, `def my_vmin(values): return\n",
      "        np.mean(values)` and then set `vmin=my_vmin`. If vmin is None (default) an automatic\n",
      "        minimum value is used as defined by matplotlib `scatter` function. When making multiple\n",
      "        plots, vmin can be a list of values, one for each plot. For example `vmin=[0.1, 'p1', None, my_vmin]`\n",
      "    vmax\n",
      "        The value representing the upper limit of the color scale. The format is the same as for `vmin`.\n",
      "    vcenter\n",
      "        The value representing the center of the color scale. Useful for diverging colormaps.\n",
      "        The format is the same as for `vmin`.\n",
      "        Example: sc.pl.umap(adata, color='TREM2', vcenter='p50', cmap='RdBu_r')\n",
      "    add_outline\n",
      "        If set to True, this will add a thin border around groups of dots. In some situations\n",
      "        this can enhance the aesthetics of the resulting image\n",
      "    outline_color\n",
      "        Tuple with two valid color names used to adjust the add_outline. The first color is the\n",
      "        border color (default: black), while the second color is a gap color between the\n",
      "        border color and the scatter dot (default: white).\n",
      "    outline_width\n",
      "        Tuple with two width numbers used to adjust the outline. The first value is the width\n",
      "        of the border color as a fraction of the scatter dot size (default: 0.3). The second value is\n",
      "        width of the gap color (default: 0.05).\n",
      "    ncols\n",
      "        Number of panels per row.\n",
      "    wspace\n",
      "        Adjust the width of the space between multiple panels.\n",
      "    hspace\n",
      "        Adjust the height of the space between multiple panels.\n",
      "    return_fig\n",
      "        Return the matplotlib figure.\n",
      "    kwargs\n",
      "        Arguments to pass to :func:`matplotlib.pyplot.scatter`,\n",
      "        for instance: the maximum and minimum values (e.g. `vmin=-2, vmax=5`).\n",
      "    show\n",
      "         Show the plot, do not return axis.\n",
      "    save\n",
      "        If `True` or a `str`, save the figure.\n",
      "        A string is appended to the default filename.\n",
      "        Infer the filetype if ending on {`'.pdf'`, `'.png'`, `'.svg'`}.\n",
      "    ax\n",
      "        A matplotlib axes object. Only works if plotting a single component.\n",
      "    \n",
      "    Returns\n",
      "    -------\n",
      "    If `show==False` a :class:`~matplotlib.axes.Axes` or a list of it.\n",
      "    \n",
      "    Examples\n",
      "    --------\n",
      "    This function behaves very similarly to other embedding plots like\n",
      "    :func:`~scanpy.pl.umap`\n",
      "    \n",
      "    >>> adata = sc.datasets.visium_sge(\"Targeted_Visium_Human_Glioblastoma_Pan_Cancer\")\n",
      "    >>> sc.pp.calculate_qc_metrics(adata, inplace=True)\n",
      "    >>> sc.pl.spatial(adata, color=\"log1p_n_genes_by_counts\")\n",
      "    \n",
      "    See Also\n",
      "    --------\n",
      "    :func:`scanpy.datasets.visium_sge`\n",
      "        Example visium data.\n",
      "    :tutorial:`spatial/basic-analysis`\n",
      "        Tutorial on spatial analysis.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(sc.pl.spatial)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68cbc7ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "rcParams[\"axes.facecolor\"] = \"black\"\n",
    "rcParams[\"figure.figsize\"] = [12,14]\n",
    "rcParams[\"font.size\"] = 28\n",
    "rcParams[\"axes.facecolor\"] = \"white\"\n",
    "\n",
    "root = 'Check/ST0/'\n",
    "if not os.path.exists(root):\n",
    "    os.makedirs(root)\n",
    "\n",
    "cmap = sns.cubehelix_palette(n_colors = 32,start = 2, rot=1.5, as_cmap = True)\n",
    "#cmap='magma'\n",
    "\n",
    "genes = [\"MBP\", 'GAD2', 'SLC17A6'] \n",
    "for gene in genes:\n",
    "    sc.pl.spatial(STs[0], \n",
    "                  color=[gene], img_key='hires', title = '',\n",
    "                  vmin=0, cmap=cmap, vmax='p99.0', show = False,\n",
    "                  gene_symbols='SYMBOL', size=1.2, colorbar_loc = None\n",
    "                 )\n",
    "    frame1 = plt.gca()\n",
    "    frame1.axes.xaxis.set_visible(False)\n",
    "    frame1.axes.yaxis.set_visible(False)\n",
    "    plt.savefig(root +'_' + gene + '.png',\n",
    "                    dpi = 128, bbox_inches='tight')\n",
    "    plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58daefee",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6c4c3242",
   "metadata": {},
   "source": [
    "# SC2Spa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d0f78e09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/win1\n",
      "/mnt/win1/C2L\n"
     ]
    }
   ],
   "source": [
    "seed_num = 2022\n",
    "\n",
    "from numpy.random import seed\n",
    "seed(seed_num)\n",
    "from tensorflow.random import set_seed\n",
    "set_seed(seed_num)\n",
    "import tensorflow as tf\n",
    "from keras.models import load_model\n",
    "\n",
    "from scipy import stats\n",
    "\n",
    "from time import time\n",
    "from IPython.display import clear_output\n",
    "import os\n",
    "\n",
    "tf.keras.utils.set_random_seed(seed_num)\n",
    "tf.config.experimental.enable_op_determinism()\n",
    "\n",
    "from SC2Spa import SI, PP, Vis, SVA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e03406ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "219.0"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_ref = STs[0]\n",
    "\n",
    "from sklearn.neighbors import NearestNeighbors\n",
    "\n",
    "coords = adata_ref.obsm['spatial']\n",
    "neigh = NearestNeighbors(n_neighbors = 1000, radius = 150)\n",
    "neigh.fit(coords)\n",
    "dis, neighbors = neigh.kneighbors(coords, 1000)\n",
    "\n",
    "dis[:, 1].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "34469a50",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "n of Referece Genes: 19375\n",
      "n of Target Genes: 25289\n",
      "n of Selected Genes: 10572\n",
      "(2983, 10572)\n",
      "(40532, 10572)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-10-19 19:21:18.811703: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2022-10-19 19:21:19.505836: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 22202 MB memory:  -> device: 0, name: NVIDIA TITAN RTX, pci bus id: 0000:3b:00.0, compute capability: 7.5\n"
     ]
    }
   ],
   "source": [
    "'''\n",
    "WD_cutoff = 0.1\n",
    "root = 'WDs/'\n",
    "i = 0\n",
    "\n",
    "adata_ref = STs[i]\n",
    "adata_query = SN\n",
    "\n",
    "#sta = time()\n",
    "#JGs, WDs = SI.WassersteinD(adata_ref, adata_query, sparse = True,\n",
    "#                           WD_cutoff = 0.1, root = 'WDs/', save = 'WDs_T' + str(i+1))\n",
    "#end = time()\n",
    "#print((end - sta) / 60.0, 'min')\n",
    "\n",
    "save = 'WDs_T' + str(i+1)\n",
    "WDs = pd.read_csv(root + save + '.csv')\n",
    "JGs = sorted(WDs[WDs['Wasserstein_Distance'] < WD_cutoff]['Gene'].tolist())\n",
    "\n",
    "seed(seed_num)\n",
    "set_seed(seed_num)\n",
    "tf.keras.utils.set_random_seed(seed_num)\n",
    "\n",
    "adata_ref = adata_ref[:, JGs]\n",
    "adata_query = adata_query[:, JGs]\n",
    "\n",
    "sta = time()\n",
    "\n",
    "neighbors, dis = SI.FineMapping(adata_ref, adata_query, sparse =True, model_path = None,\n",
    "                                WD_cutoff = None, root = 'Model_SI/', name = 'SI_T' + str(i),\n",
    "                                l1_reg = 1e-5, l2_reg = 0, dropout = 0.05, epoch = 500, batch_size = 128,\n",
    "                                nodes = [4096, 1024, 256, 64, 16, 4], lrr_patience = 20, ES_patience = 50,\n",
    "                                min_lr = 1e-5, save = True, polar = True, n_neighbors = 1000,\n",
    "                                dis_cutoff = 110, seed = seed_num)\n",
    "\n",
    "end = time()\n",
    "print((end - sta) / 60.0, 'min')\n",
    "\n",
    "'''\n",
    "\n",
    "i = 0\n",
    "\n",
    "WD_cutoff = 0.1\n",
    "root = 'WDs/'\n",
    "save = 'WDs_T' + str(i+1)\n",
    "WDs = pd.read_csv(root + save + '.csv')\n",
    "JGs = sorted(WDs[WDs['Wasserstein_Distance'] < WD_cutoff]['Gene'].tolist())\n",
    "\n",
    "adata_ref = STs[i]\n",
    "adata_query = SN\n",
    "\n",
    "#adata_ref = adata_ref[:, JGs]\n",
    "#adata_query = adata_query[:, JGs]\n",
    "\n",
    "neighbors, dis = SI.FineMapping(adata_ref, adata_query, sparse =True, JGs = JGs,\n",
    "                                model_path = 'Model_SI/SI_T' + str(i) + '.h5', WD_cutoff = None, \n",
    "                                polar = True, n_neighbors = 1000, dis_cutoff = 110)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e104964",
   "metadata": {},
   "source": [
    "\n",
    "# Visualize Cell Type Locations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14491f20",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "individual1 = adata_query.obs['sample'].apply(lambda x: x in ['5705STDY8058280', \n",
    "                                                             '5705STDY8058281',\n",
    "                                                             '5705STDY8058282']).tolist()\n",
    "\n",
    "for CT in adata_query.obs['annotation_1'].value_counts().index.tolist():\n",
    "    print('*'*16)\n",
    "    print(CT)\n",
    "    print('*'*16)\n",
    "    print('Transfer:')\n",
    "    plt.figure(figsize = (12, 12))\n",
    "\n",
    "    plt.imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "    Vis.DrawCT1(adata_query[individual1], coords_name='spatial_mapping_hires', CT = CT, s = 8, FM = True, \n",
    "                c_name = 'annotation_1', root = 'Transfer' + str(i+1) +'/FM_Valid1/',\n",
    "                figsize = None, save = 'T' + str(i+1) + '_individual1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17fe1fd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_row = 12\n",
    "n_col = 5\n",
    "i = 0\n",
    "root = 'Transfer' + str(i+1) +'/FM_Valid1/'\n",
    "save = 'T' + str(i+1) + '_individual1'\n",
    "\n",
    "fig, axes = plt.subplots(n_row, n_col, figsize = (6*n_col, 6*n_row))\n",
    "\n",
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "individual1 = adata_query.obs['sample'].apply(lambda x: x in ['5705STDY8058280', \n",
    "                                                              '5705STDY8058281',\n",
    "                                                              '5705STDY8058282']).tolist()\n",
    "\n",
    "for i, CT in enumerate(adata_query.obs['annotation_1'].value_counts().index.tolist()):\n",
    "    \n",
    "    i_row = int(i/n_col)\n",
    "    i_col = int(i%n_col)\n",
    "    print(i_row, i_col)\n",
    "    print('*'*16)\n",
    "    print(CT)\n",
    "    print('*'*16)\n",
    "    print('Transfer:')\n",
    "\n",
    "    axes[i_row][i_col].imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "    DrawCT1(adata_query[individual1], coords_name='spatial_mapping_hires',\n",
    "                ax = axes[i_row][i_col], CT = CT, s = 2, FM = True, c_name = 'annotation_1')\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.savefig(root + save + '_all' + '.png',\n",
    "            dpi=128, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47cae211",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "individual2 = adata_query.obs['sample'].apply(lambda x: x in ['5705STDY8058283', \n",
    "                                                             '5705STDY8058284',\n",
    "                                                             '5705STDY8058285']).tolist()\n",
    "\n",
    "for CT in adata_query.obs['annotation_1'].value_counts().index.tolist():\n",
    "    print('*'*16)\n",
    "    print(CT)\n",
    "    print('*'*16)\n",
    "    print('Transfer:')\n",
    "    plt.figure(figsize = (12, 12))\n",
    "\n",
    "    plt.imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "    Vis.DrawCT1(adata_query[individual2], coords_name='spatial_mapping_hires', CT = CT, s = 8, FM = True, \n",
    "                c_name = 'annotation_1', root = 'Transfer' + str(i+1) +'/FM_Valid1/',\n",
    "                figsize = None, save = 'T' + str(i+1) + '_individual2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed550e99",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_row = 12\n",
    "n_col = 5\n",
    "i=0\n",
    "\n",
    "root = 'Transfer' + str(i+1) +'/FM_Valid1/'\n",
    "save = 'T' + str(i+1) + '_individual2'\n",
    "\n",
    "fig, axes = plt.subplots(n_row, n_col, figsize = (6*n_col, 6*n_row))\n",
    "\n",
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "individual2 = adata_query.obs['sample'].apply(lambda x: x in ['5705STDY8058283', \n",
    "                                                             '5705STDY8058284',\n",
    "                                                             '5705STDY8058285']).tolist()\n",
    "\n",
    "for i, CT in enumerate(adata_query.obs['annotation_1'].value_counts().index.tolist()):\n",
    "    i_row = int(i/n_col)\n",
    "    i_col = int(i%n_col)\n",
    "    print(i_row, i_col)\n",
    "    print('*'*16)\n",
    "    print(CT)\n",
    "    print('*'*16)\n",
    "    print('Transfer:')\n",
    "\n",
    "    axes[i_row][i_col].imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "    DrawCT1(adata_query[individual2], coords_name='spatial_mapping_hires',\n",
    "                ax = axes[i_row][i_col], CT = CT, s = 2, FM = True, c_name = 'annotation_1')\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.savefig(root + save + '_all' + '.png',\n",
    "                        dpi=128, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd90a44b",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "\n",
    "for CT in adata_query.obs['annotation_1'].value_counts().index.tolist():\n",
    "    print('*'*16)\n",
    "    print(CT)\n",
    "    print('*'*16)\n",
    "    print('Transfer:')\n",
    "    plt.figure(figsize = (12, 12))\n",
    "\n",
    "    plt.imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "    Vis.DrawCT1(adata_query, coords_name='spatial_mapping_hires', CT = CT, s = 2, FM = True, \n",
    "                c_name = 'annotation_1', root = 'Transfer' + str(i+1) +'/FM_Valid1/',\n",
    "                figsize = None, save = 'T' + str(i+1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5252e0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize = (12, 12))\n",
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "coords =adata_ref.obsm['spatial_mapping'] * scalef\n",
    "\n",
    "plt.imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "plt.scatter(coords[:, 0], coords[:, 1], s = 12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b93ac522",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2700, 1000)\n",
      "(2983, 19375)\n",
      "(3495, 19268)\n",
      "(3492, 19587)\n",
      "(2409, 18966)\n",
      "(2575, 19306)\n"
     ]
    }
   ],
   "source": [
    "print(neighbors.shape)\n",
    "for i in range(5):\n",
    "    print(STs[i].shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e152f71d",
   "metadata": {},
   "source": [
    "# Impute"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c2062b90",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.2869393706321716 min(s)\n"
     ]
    }
   ],
   "source": [
    "sta = time()\n",
    "\n",
    "#SN is adata_query before selecting genes\n",
    "adata_impute = SI.NRD_impute(neighbors, dis, adata_ref, SN,\n",
    "                             ct_name='annotation_1', dis_min=1, exclude_CTs=None)\n",
    "\n",
    "end = time()\n",
    "print((end - sta) / 60, 'min(s)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "efdfc4d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs  n_vars = 2700  25289\n",
       "    obs: 'in_tissue', 'array_row', 'array_col', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'FM'\n",
       "    obsm: 'spatial'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_impute"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "133a2740",
   "metadata": {},
   "source": [
    "# SVA"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8221f2eb",
   "metadata": {},
   "source": [
    "## Moran's I on Single Cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "31dfaff6",
   "metadata": {},
   "outputs": [],
   "source": [
    "import squidpy as sq\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eef8b752",
   "metadata": {},
   "outputs": [],
   "source": [
    "CTs = adata_query.obs['annotation_1'].value_counts()\n",
    "\n",
    "CTs_selected = CTs[CTs > 200].index.tolist()\n",
    "print(CTs[CTs > 200])\n",
    "#CTs_selected = CTs[CTs.index.str.contains('Astro')].index.tolist()\n",
    "#print(CTs_selected)\n",
    "\n",
    "root = 'MoransI/'\n",
    "if not os.path.exists(root):\n",
    "    os.makedirs(root)\n",
    "\n",
    "sq.gr.spatial_neighbors(adata_query, spatial_key = 'spatial_mapping')\n",
    "    \n",
    "result_CTS = []\n",
    "i = 0\n",
    "for CT in CTs_selected:\n",
    "    print(CT)\n",
    "    \n",
    "    '''\n",
    "    if(i==0):\n",
    "        i=1\n",
    "        print('PASS')\n",
    "        continue\n",
    "    '''\n",
    "    t = adata_query[adata_query.obs['annotation_1'] == CT]\n",
    "    sq.gr.spatial_autocorr(\n",
    "        t,\n",
    "        mode=\"moran\",\n",
    "        n_perms=100,\n",
    "        n_jobs=16,\n",
    "    )\n",
    "    t.uns[\"moranI\"].to_csv('MoransI/MI_' + CT + '.csv')\n",
    "    result_CTS.append(t.uns[\"moranI\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "bca524b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2d242328d03643f8841a9ea5e6f4f342",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/100 [00:00<?, ?/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "/home/linbuliao/anaconda3/lib/python3.9/site-packages/scanpy/metrics/_gearys_c.py:293: UserWarning: 1125 variables were constant, will return nan for these.\n",
      "  warnings.warn(\n",
      "Trying to set attribute `._uns` of view, copying.\n"
     ]
    }
   ],
   "source": [
    "###Astrocyte\n",
    "root = 'MoransI/'\n",
    "if not os.path.exists(root):\n",
    "    os.makedirs(root)\n",
    "\n",
    "sq.gr.spatial_neighbors(adata_query, spatial_key = 'spatial_mapping')\n",
    "    \n",
    "\n",
    "t = adata_query[adata_query.obs['annotation_1'].str.contains('Astro')]\n",
    "sq.gr.spatial_autocorr(\n",
    "    t,\n",
    "    mode=\"moran\",\n",
    "    n_perms=100,\n",
    "    n_jobs=16,\n",
    ")\n",
    "t.uns[\"moranI\"].to_csv('MoransI/MI_Astro.csv')\n",
    "result_Astro = t.uns[\"moranI\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "fd62f709",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(10572,)"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = adata_query[adata_query.obs['annotation_1'].str.contains('Astro')]\n",
    "t.X.toarray().var(axis = 0).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "2bf1545b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>I</th>\n",
       "      <th>pval_norm</th>\n",
       "      <th>var_norm</th>\n",
       "      <th>pval_z_sim</th>\n",
       "      <th>pval_sim</th>\n",
       "      <th>var_sim</th>\n",
       "      <th>pval_norm_fdr_bh</th>\n",
       "      <th>pval_z_sim_fdr_bh</th>\n",
       "      <th>pval_sim_fdr_bh</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCOM1</th>\n",
       "      <td>0.672708</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0.004392</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPX</th>\n",
       "      <td>0.552215</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0.000755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCGF2</th>\n",
       "      <td>0.397710</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0.001447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIFC1</th>\n",
       "      <td>0.397105</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0.001068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EDN1</th>\n",
       "      <td>0.396671</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0.000170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP819</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP87</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP986</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP992</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZP3R</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04603</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10572 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       "GCOM1   0.672708        0.0  0.000937         0.0  0.009901  0.004392   \n",
       "SPX     0.552215        0.0  0.000937         0.0  0.009901  0.000755   \n",
       "PCGF2   0.397710        0.0  0.000937         0.0  0.009901  0.001447   \n",
       "KIFC1   0.397105        0.0  0.000937         0.0  0.009901  0.001068   \n",
       "EDN1    0.396671        0.0  0.000937         0.0  0.009901  0.000170   \n",
       "...          ...        ...       ...         ...       ...       ...   \n",
       "ZFP819       NaN        NaN  0.000937         NaN  0.009901       NaN   \n",
       "ZFP87        NaN        NaN  0.000937         NaN  0.009901       NaN   \n",
       "ZFP986       NaN        NaN  0.000937         NaN  0.009901       NaN   \n",
       "ZFP992       NaN        NaN  0.000937         NaN  0.009901       NaN   \n",
       "ZP3R         NaN        NaN  0.000937         NaN  0.009901       NaN   \n",
       "\n",
       "        pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       "GCOM1                NaN                NaN          0.04603  \n",
       "SPX                  NaN                NaN          0.04603  \n",
       "PCGF2                NaN                NaN          0.04603  \n",
       "KIFC1                NaN                NaN          0.04603  \n",
       "EDN1                 NaN                NaN          0.04603  \n",
       "...                  ...                ...              ...  \n",
       "ZFP819               NaN                NaN          0.04603  \n",
       "ZFP87                NaN                NaN          0.04603  \n",
       "ZFP986               NaN                NaN          0.04603  \n",
       "ZFP992               NaN                NaN          0.04603  \n",
       "ZP3R                 NaN                NaN          0.04603  \n",
       "\n",
       "[10572 rows x 9 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_Astro.sort_values('I', ascending = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8049dc78",
   "metadata": {},
   "source": [
    "## Visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5cf7105a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Astro_AMY         498\n",
       "Astro_HYPO        451\n",
       "Astro_CTX         443\n",
       "Astro_THAL_med    405\n",
       "Astro_THAL_lat    332\n",
       "Astro_AMY_CTX     279\n",
       "Astro_HPC         261\n",
       "Astro_WM          220\n",
       "Astro_STR          81\n",
       "Astro_THAL_hab     43\n",
       "Name: annotation_1, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CTs[CTs.index.str.contains('Astro')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "4cad9690",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0740101585282594"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MI_subtype = pd.read_csv('MoransI/MI_Astro_CTX.csv', index_col = 0)\n",
    "MI_subtype[\"var_norm\"].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec8ec4e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "\n",
    "cmap = sns.cubehelix_palette(n_colors = 32,start = 2, rot=1.5, as_cmap = True)\n",
    "\n",
    "s1 = 27\n",
    "s2 = 54\n",
    "i = 0\n",
    "\n",
    "CT = 'Astro'\n",
    "print('*'*16)\n",
    "print(CT)\n",
    "print('*'*16)\n",
    "MI_subtype = pd.read_csv('MoransI/MI_' + CT + '.csv', index_col = 0)\n",
    "\n",
    "Genes = MI_subtype.sort_values('I', ascending = False)[:3]['I'].index.tolist()\n",
    "\n",
    "for gene in Genes:\n",
    "    print('*'*32)\n",
    "    print(gene)\n",
    "    print('*'*32)\n",
    "    print('scRNAseq:')\n",
    "    plt.figure(figsize = (12, 12))\n",
    "\n",
    "    plt.imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "    plt.title(CT)\n",
    "    Vis.DrawGenes2(adata_query[adata_query.obs['annotation_1'].str.contains('Astro')],\n",
    "                   coords_name = 'spatial_mapping_hires', gene = gene, lim = False, cmap = cmap,\n",
    "                   FM = True, CTL = None, c_name = 'annotation_1', root = 'Transfer' + str(i+1) + '/SVA/' + CT + '/',\n",
    "                   figsize = None, s = s1, title = False, save = 'SC')\n",
    "    print('ST:')\n",
    "    Vis.DrawGenes2(adata_ref, coords_name = 'spatial', gene = gene, lim = False, cmap = cmap,\n",
    "                   FM = True, CTL = None, c_name = 'annotation_1', root = 'Transfer' + str(i+1) + '/SVA/' + CT + '/', \n",
    "                   s = s2, title = False, save = 'ST' + str(i+1) + '_' + gene)\n",
    "    print('Imputed ST:')\n",
    "    Vis.DrawGenes2(adata_impute, coords_name = 'spatial', gene = gene, lim = False, cmap = cmap,\n",
    "                   FM = True, CTL = None, c_name = 'annotation_1', root = 'Transfer' + str(i+1) + '/SVA/' + CT + '/', \n",
    "                   s = s2, title = False, save = 'ST' + str(i+1) + '_' + gene + '_Impute')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5c62372",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_id = list(adata_ref.uns['spatial'].keys())[0]\n",
    "scalef = adata_ref.uns['spatial'][lib_id]['scalefactors']['tissue_hires_scalef']\n",
    "adata_query.obsm['spatial_mapping_hires'] =adata_query.obsm['spatial_mapping'] * scalef\n",
    "\n",
    "cmap = sns.cubehelix_palette(n_colors = 32,start = 2, rot=1.5, as_cmap = True)\n",
    "\n",
    "s1 = 54\n",
    "s2 = 54\n",
    "i = 0\n",
    "\n",
    "\n",
    "CTs = adata_query.obs['annotation_1'].value_counts()\n",
    "CTs_selected = CTs[CTs.index.str.contains('Astro')].index.tolist()\n",
    "\n",
    "for CT in CTs_selected:\n",
    "    \n",
    "    print('*'*16)\n",
    "    print(CT)\n",
    "    print('*'*16)\n",
    "    MI_subtype = pd.read_csv('MoransI/MI_' + CT + '.csv', index_col = 0)\n",
    "\n",
    "    Genes = MI_subtype.sort_values('I', ascending = False)[:3]['I'].index.tolist()\n",
    "\n",
    "    for gene in Genes:\n",
    "        print('*'*32)\n",
    "        print(gene)\n",
    "        print('*'*32)\n",
    "        print('scRNAseq:')\n",
    "        plt.figure(figsize = (12, 12))\n",
    "\n",
    "        plt.imshow(adata_ref.uns['spatial'][lib_id]['images']['hires'])\n",
    "        plt.title(CT)\n",
    "        Vis.DrawGenes2(adata_query,coords_name = 'spatial_mapping_hires', gene = gene, lim = False, cmap = cmap,\n",
    "                       FM = True, CTL = [CT], c_name = 'annotation_1', root = 'Transfer' + str(i+1) + '/SVA/' + CT + '/',\n",
    "                       figsize = None, s = s1, title = False, save = 'SC')\n",
    "        print('ST:')\n",
    "        Vis.DrawGenes2(adata_ref, coords_name = 'spatial', gene = gene, lim = False, cmap = cmap,\n",
    "                       FM = True, CTL = None, c_name = 'annotation_1', root = 'Transfer' + str(i+1) + '/SVA/' + CT + '/', \n",
    "                       s = s2, title = False, save = 'ST' + str(i+1) + '_' + gene)\n",
    "        print('Imputed ST:')\n",
    "        Vis.DrawGenes2(adata_impute, coords_name = 'spatial', gene = gene, lim = False, cmap = cmap,\n",
    "                       FM = True, CTL = None, c_name = 'annotation_1', root = 'Transfer' + str(i+1) + '/SVA/' + CT + '/', \n",
    "                       s = s2, title = False, save = 'ST' + str(i+1) + '_' + gene + '_Impute')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "075f7c22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs  n_vars = 40532  10572\n",
       "    obs: 'Cell ID', 'batch', 'sample', 'annotation_1', 'annotation_1_print', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'FM'\n",
       "    var: 'gene_ids', 'feature_types', 'genome', 'n_cells', 'mt', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts'\n",
       "    uns: 'log1p', 'pca', 'annotation_1_colors', 'neighbors', 'umap', 'spatial_neighbors', 'moranI'\n",
       "    obsm: 'X_pca', 'X_umap', 'spatial_mapping'\n",
       "    varm: 'PCs'\n",
       "    obsp: 'distances', 'connectivities', 'spatial_connectivities', 'spatial_distances'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "51c13f37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[                      I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000621  0.019802  0.000004   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000425  0.009901  0.000005   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000235  0.009901  0.000005   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000115  0.009901  0.000005   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024422  \n",
       " ABHD12B                     NaN                NaN         0.024422  \n",
       " A830036E02RIK               NaN                NaN         0.024422  \n",
       " ADORA2A                     NaN                NaN         0.024422  \n",
       " GM35696                     NaN                NaN         0.024422  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.043705  \n",
       " MYL6B                       NaN                NaN         0.024422  \n",
       " ZFP213                      NaN                NaN         0.024422  \n",
       " GM12359                     NaN                NaN         0.024422  \n",
       " GM27168                     NaN                NaN         0.024422  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000010   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000563  0.009901  0.000005   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.001130  0.009901  0.000006   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000032  0.009901  0.000004   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000364  0.009901  0.000005   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024303  \n",
       " ABHD12B                     NaN                NaN         0.024303  \n",
       " A830036E02RIK               NaN                NaN         0.024303  \n",
       " ADORA2A                     NaN                NaN         0.024303  \n",
       " GM35696                     NaN                NaN         0.024303  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024303  \n",
       " MYL6B                       NaN                NaN         0.024303  \n",
       " ZFP213                      NaN                NaN         0.024303  \n",
       " GM12359                     NaN                NaN         0.024303  \n",
       " GM27168                     NaN                NaN         0.024303  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000011   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000010   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000813  0.009901  0.000005   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000057  0.009901  0.000004   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000092  0.009901  0.000004   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000095  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024371  \n",
       " ABHD12B                     NaN                NaN         0.024371  \n",
       " A830036E02RIK               NaN                NaN         0.024371  \n",
       " ADORA2A                     NaN                NaN         0.024371  \n",
       " GM35696                     NaN                NaN         0.024371  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024371  \n",
       " MYL6B                       NaN                NaN         0.024371  \n",
       " ZFP213                      NaN                NaN         0.024371  \n",
       " GM12359                     NaN                NaN         0.024371  \n",
       " GM27168                     NaN                NaN         0.024371  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000938  0.009901  0.000005   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000261  0.009901  0.000005   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000073  0.009901  0.000004   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000047  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024343  \n",
       " ABHD12B                     NaN                NaN         0.024343  \n",
       " A830036E02RIK               NaN                NaN         0.024343  \n",
       " ADORA2A                     NaN                NaN         0.024343  \n",
       " GM35696                     NaN                NaN         0.024343  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024343  \n",
       " MYL6B                       NaN                NaN         0.024343  \n",
       " ZFP213                      NaN                NaN         0.024343  \n",
       " GM12359                     NaN                NaN         0.024343  \n",
       " GM27168                     NaN                NaN         0.024343  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000010   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000010   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000621  0.009901  0.000004   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000074  0.009901  0.000004   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000615  0.009901  0.000005   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000032  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024348  \n",
       " ABHD12B                     NaN                NaN         0.024348  \n",
       " A830036E02RIK               NaN                NaN         0.024348  \n",
       " ADORA2A                     NaN                NaN         0.024348  \n",
       " GM35696                     NaN                NaN         0.024348  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024348  \n",
       " MYL6B                       NaN                NaN         0.024348  \n",
       " ZFP213                      NaN                NaN         0.024348  \n",
       " GM12359                     NaN                NaN         0.024348  \n",
       " GM27168                     NaN                NaN         0.024348  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000010   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.001722  0.009901  0.000005   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000203  0.009901  0.000004   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000307  0.009901  0.000005   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000195  0.009901  0.000005   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN          0.02418  \n",
       " ABHD12B                     NaN                NaN          0.02418  \n",
       " A830036E02RIK               NaN                NaN          0.02418  \n",
       " ADORA2A                     NaN                NaN          0.02418  \n",
       " GM35696                     NaN                NaN          0.02418  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN          0.02418  \n",
       " MYL6B                       NaN                NaN          0.02418  \n",
       " ZFP213                      NaN                NaN          0.02418  \n",
       " GM12359                     NaN                NaN          0.02418  \n",
       " GM27168                     NaN                NaN          0.02418  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000416  0.009901  0.000004   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000280  0.009901  0.000005   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000430  0.009901  0.000005   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000036  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024264  \n",
       " ABHD12B                     NaN                NaN         0.024264  \n",
       " A830036E02RIK               NaN                NaN         0.024264  \n",
       " ADORA2A                     NaN                NaN         0.024264  \n",
       " GM35696                     NaN                NaN         0.024264  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024264  \n",
       " MYL6B                       NaN                NaN         0.024264  \n",
       " ZFP213                      NaN                NaN         0.024264  \n",
       " GM12359                     NaN                NaN         0.024264  \n",
       " GM27168                     NaN                NaN         0.024264  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000009   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000006   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000324  0.009901  0.000004   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000512  0.009901  0.000005   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000044  0.009901  0.000004   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000137  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024348  \n",
       " ABHD12B                     NaN                NaN         0.024348  \n",
       " A830036E02RIK               NaN                NaN         0.024348  \n",
       " ADORA2A                     NaN                NaN         0.024348  \n",
       " GM35696                     NaN                NaN         0.024348  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024348  \n",
       " MYL6B                       NaN                NaN         0.024348  \n",
       " ZFP213                      NaN                NaN         0.024348  \n",
       " GM12359                     NaN                NaN         0.024348  \n",
       " GM27168                     NaN                NaN         0.024348  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000013   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000011   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000006   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.000656  0.009901  0.000004   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000524  0.009901  0.000005   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000028  0.009901  0.000004   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000051  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024411  \n",
       " ABHD12B                     NaN                NaN         0.024411  \n",
       " A830036E02RIK               NaN                NaN         0.024411  \n",
       " ADORA2A                     NaN                NaN         0.024411  \n",
       " GM35696                     NaN                NaN         0.024411  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024411  \n",
       " MYL6B                       NaN                NaN         0.024411  \n",
       " ZFP213                      NaN                NaN         0.024411  \n",
       " GM12359                     NaN                NaN         0.024411  \n",
       " GM27168                     NaN                NaN         0.024411  \n",
       " \n",
       " [10572 rows x 9 columns],\n",
       "                       I  pval_norm  var_norm  pval_z_sim  pval_sim   var_sim  \\\n",
       " RGS9           0.219550   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ABHD12B        0.216440   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " A830036E02RIK  0.194917   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ADORA2A        0.185095   0.000000  0.000008    0.000000  0.009901  0.000007   \n",
       " GM35696        0.163380   0.000000  0.000008    0.000000  0.009901  0.000008   \n",
       " ...                 ...        ...       ...         ...       ...       ...   \n",
       " GM17484       -0.006671   0.010956  0.000008    0.001575  0.009901  0.000005   \n",
       " MYL6B         -0.007465   0.005149  0.000008    0.000074  0.009901  0.000004   \n",
       " ZFP213        -0.007533   0.004810  0.000008    0.000512  0.009901  0.000005   \n",
       " GM12359       -0.007846   0.003496  0.000008    0.000113  0.009901  0.000004   \n",
       " GM27168             NaN        NaN  0.000008         NaN  0.009901       NaN   \n",
       " \n",
       "                pval_norm_fdr_bh  pval_z_sim_fdr_bh  pval_sim_fdr_bh  \n",
       " RGS9                        NaN                NaN         0.024264  \n",
       " ABHD12B                     NaN                NaN         0.024264  \n",
       " A830036E02RIK               NaN                NaN         0.024264  \n",
       " ADORA2A                     NaN                NaN         0.024264  \n",
       " GM35696                     NaN                NaN         0.024264  \n",
       " ...                         ...                ...              ...  \n",
       " GM17484                     NaN                NaN         0.024264  \n",
       " MYL6B                       NaN                NaN         0.024264  \n",
       " ZFP213                      NaN                NaN         0.024264  \n",
       " GM12359                     NaN                NaN         0.024264  \n",
       " GM27168                     NaN                NaN         0.024264  \n",
       " \n",
       " [10572 rows x 9 columns]]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_CTS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43055fba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e07153d3",
   "metadata": {},
   "source": [
    "## SC2Spa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "a3fba2a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<keras.layers.core.dense.Dense object at 0x7f78685cd370> :\n",
      "(4, 2)\n",
      "<keras.layers.core.dense.Dense object at 0x7f78635b7ee0> :\n",
      "(16, 4)\n",
      "<keras.layers.core.dense.Dense object at 0x7f7818548c10> :\n",
      "(64, 16)\n",
      "<keras.layers.core.dense.Dense object at 0x7f78686d5a60> :\n",
      "(256, 64)\n",
      "<keras.layers.core.dense.Dense object at 0x7f78619ac940> :\n",
      "(1024, 256)\n",
      "<keras.layers.core.dense.Dense object at 0x7f76f3febdc0> :\n",
      "(4096, 1024)\n",
      "<keras.layers.core.dense.Dense object at 0x7f786f66fc10> :\n",
      "(10572, 4096)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.var` of view, copying.\n"
     ]
    }
   ],
   "source": [
    "i = 0\n",
    "JGs = sorted(list(set(adata_ref.var_names).intersection(set(adata_query.var_names))))\n",
    "\n",
    "adata = adata_ref[:, JGs]\n",
    "adata.var['gene'] = adata.var.index.tolist()\n",
    "\n",
    "model = load_model('Model_SI/SI_T' + str(i) + '.h5', compile=False)\n",
    "SVA.PrioritizeLPG(adata, Model = model, sparse=False, percent = 0.5, scale_factor = 1e3,\n",
    "                  Norm = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "935c6372",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "      <th>genome</th>\n",
       "      <th>n_cells</th>\n",
       "      <th>mt</th>\n",
       "      <th>n_cells_by_counts</th>\n",
       "      <th>mean_counts</th>\n",
       "      <th>pct_dropout_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>imp_sumup</th>\n",
       "      <th>imp_sumup_norm</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SPINK8</th>\n",
       "      <td>ENSMUSG00000050074</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0_premrna</td>\n",
       "      <td>386</td>\n",
       "      <td>False</td>\n",
       "      <td>386</td>\n",
       "      <td>0.355682</td>\n",
       "      <td>87.060007</td>\n",
       "      <td>1061.0</td>\n",
       "      <td>1.528409</td>\n",
       "      <td>2.371915</td>\n",
       "      <td>SPINK8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CYP26B1</th>\n",
       "      <td>ENSMUSG00000063415</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0_premrna</td>\n",
       "      <td>346</td>\n",
       "      <td>False</td>\n",
       "      <td>346</td>\n",
       "      <td>0.213208</td>\n",
       "      <td>88.400939</td>\n",
       "      <td>636.0</td>\n",
       "      <td>1.232464</td>\n",
       "      <td>1.912642</td>\n",
       "      <td>CYP26B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A830009L08RIK</th>\n",
       "      <td>ENSMUSG00000097182</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0_premrna</td>\n",
       "      <td>294</td>\n",
       "      <td>False</td>\n",
       "      <td>294</td>\n",
       "      <td>0.116996</td>\n",
       "      <td>90.144150</td>\n",
       "      <td>349.0</td>\n",
       "      <td>1.131165</td>\n",
       "      <td>1.755438</td>\n",
       "      <td>A830009L08RIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C1QL2</th>\n",
       "      <td>ENSMUSG00000036907</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0_premrna</td>\n",
       "      <td>355</td>\n",
       "      <td>False</td>\n",
       "      <td>355</td>\n",
       "      <td>0.359035</td>\n",
       "      <td>88.099229</td>\n",
       "      <td>1071.0</td>\n",
       "      <td>1.092777</td>\n",
       "      <td>1.695864</td>\n",
       "      <td>C1QL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STARD8</th>\n",
       "      <td>ENSMUSG00000031216</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>mm10-3.0.0_premrna</td>\n",
       "      <td>529</td>\n",
       "      <td>False</td>\n",
       "      <td>529</td>\n",
       "      <td>0.257459</td>\n",
       "      <td>82.266175</td>\n",
       "      <td>768.0</td>\n",
       "      <td>0.996261</td>\n",
       "      <td>1.546082</td>\n",
       "      <td>STARD8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         gene_ids    feature_types              genome  \\\n",
       "SPINK8         ENSMUSG00000050074  Gene Expression  mm10-3.0.0_premrna   \n",
       "CYP26B1        ENSMUSG00000063415  Gene Expression  mm10-3.0.0_premrna   \n",
       "A830009L08RIK  ENSMUSG00000097182  Gene Expression  mm10-3.0.0_premrna   \n",
       "C1QL2          ENSMUSG00000036907  Gene Expression  mm10-3.0.0_premrna   \n",
       "STARD8         ENSMUSG00000031216  Gene Expression  mm10-3.0.0_premrna   \n",
       "\n",
       "               n_cells     mt  n_cells_by_counts  mean_counts  \\\n",
       "SPINK8             386  False                386     0.355682   \n",
       "CYP26B1            346  False                346     0.213208   \n",
       "A830009L08RIK      294  False                294     0.116996   \n",
       "C1QL2              355  False                355     0.359035   \n",
       "STARD8             529  False                529     0.257459   \n",
       "\n",
       "               pct_dropout_by_counts  total_counts  imp_sumup  imp_sumup_norm  \\\n",
       "SPINK8                     87.060007        1061.0   1.528409        2.371915   \n",
       "CYP26B1                    88.400939         636.0   1.232464        1.912642   \n",
       "A830009L08RIK              90.144150         349.0   1.131165        1.755438   \n",
       "C1QL2                      88.099229        1071.0   1.092777        1.695864   \n",
       "STARD8                     82.266175         768.0   0.996261        1.546082   \n",
       "\n",
       "                        gene  \n",
       "SPINK8                SPINK8  \n",
       "CYP26B1              CYP26B1  \n",
       "A830009L08RIK  A830009L08RIK  \n",
       "C1QL2                  C1QL2  \n",
       "STARD8                STARD8  "
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.var.sort_values('imp_sumup_norm', ascending = False).head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed142c53",
   "metadata": {},
   "source": [
    "# Pearson's r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "9e1507cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "cmap = sns.cubehelix_palette(n_colors = 32,start = 2, rot=1.5, as_cmap = True)\n",
    "\n",
    "def VizCus(df, x_name, x_label, y_name, y_label, c_name = None,\n",
    "           fontsize = 28, vmin= -0.05, vmax = 0.5, s = 14, \n",
    "           root = 'transfer1/', save = None):\n",
    "    \n",
    "    plt.rcParams['font.size'] = fontsize\n",
    "    plt.figure(figsize = (12, 8))\n",
    "    if(c_name == None):\n",
    "        plt.scatter(df[x_name], df[y_name], s = s)\n",
    "    elif(vmin!=None):\n",
    "        plt.scatter(df[x_name], df[y_name], s = s, alpha = 0.7,\n",
    "                c = df[c_name], vmin = vmin, vmax = vmax, cmap = cmap)\n",
    "    else:\n",
    "        plt.scatter(df[x_name], df[y_name], s = s, alpha = 0.7,\n",
    "                c = df[c_name], cmap = cmap)\n",
    "    plt.colorbar()\n",
    "    plt.xlabel(x_label)\n",
    "    plt.ylabel(y_label)\n",
    "    \n",
    "    if(save != None):\n",
    "        if not os.path.exists(root):\n",
    "            os.makedirs(root)\n",
    "        plt.savefig(root + save + '.png',\n",
    "                    dpi = 128, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "845ce3d9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "189\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_2081338/4179328208.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mJG\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mJGs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0mt\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpearsonr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0madata_impute\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mJCs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mJG\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflatten\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0madata_ref\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mJCs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mJG\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtoarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflatten\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m     \u001b[0mpearson_JGs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'pearson_r'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0mpearson_JGs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'p'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/anndata/_core/anndata.py\u001b[0m in \u001b[0;36mX\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    623\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_view\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    624\u001b[0m             X = as_view(\n\u001b[0;32m--> 625\u001b[0;31m                 \u001b[0m_subset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_adata_ref\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_oidx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_vidx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    626\u001b[0m                 \u001b[0mElementRef\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"X\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    627\u001b[0m             )\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/functools.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kw)\u001b[0m\n\u001b[1;32m    875\u001b[0m                             '1 positional argument')\n\u001b[1;32m    876\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 877\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mdispatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__class__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    878\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    879\u001b[0m     \u001b[0mfuncname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'__name__'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'singledispatch function'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/anndata/_core/index.py\u001b[0m in \u001b[0;36m_subset_spmatrix\u001b[0;34m(a, subset_idx)\u001b[0m\n\u001b[1;32m    133\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset_idx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcabc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mIterable\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msubset_idx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    134\u001b[0m         \u001b[0msubset_idx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0msubset_idx\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0msubset_idx\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 135\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msubset_idx\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    136\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    137\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/scipy/sparse/_index.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m     55\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_arrayXint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcol\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     56\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mslice\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 57\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_arrayXslice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcol\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     58\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# row.ndim == 2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mINT_TYPES\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/scipy/sparse/csr.py\u001b[0m in \u001b[0;36m_get_arrayXslice\u001b[0;34m(self, row, col)\u001b[0m\n\u001b[1;32m    330\u001b[0m             \u001b[0mcol\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mcol\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindices\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    331\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_arrayXarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcol\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 332\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_major_index_fancy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_submatrix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mminor\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcol\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    333\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    334\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/scipy/sparse/compressed.py\u001b[0m in \u001b[0;36m_get_submatrix\u001b[0;34m(self, major, minor, copy)\u001b[0m\n\u001b[1;32m    789\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mcopy\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    790\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 791\u001b[0;31m         indptr, indices, data = get_csr_submatrix(\n\u001b[0m\u001b[1;32m    792\u001b[0m             M, N, self.indptr, self.indices, self.data, i0, i1, j0, j1)\n\u001b[1;32m    793\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "JCs = sorted(list(set(adata_ref.obs_names).intersection(set(adata_impute.obs_names))))\n",
    "\n",
    "pearson_JGs = pd.DataFrame(np.zeros((len(JGs), 3)), columns = ['gene', 'pearson_r', 'p']) \n",
    "pearson_JGs['gene'] = JGs\n",
    "\n",
    "for i, JG in enumerate(JGs):\n",
    "    print(i)\n",
    "    t = stats.pearsonr(adata_impute[JCs, JG].X.flatten(), adata_ref[JCs, JG].X.toarray().flatten())\n",
    "    pearson_JGs.loc[i, 'pearson_r'] = t[0]\n",
    "    pearson_JGs.loc[i, 'p'] = t[1]\n",
    "    clear_output(wait = True)\n",
    "    \n",
    "sc.pp.calculate_qc_metrics(adata, percent_top=None, log1p=False, inplace=True)\n",
    "adata.var['percent_cell'] = adata.var['n_cells_by_counts'] / adata.shape[0]\n",
    "\n",
    "t = adata.var[['percent_cell', 'imp_sumup_norm']].reset_index()\n",
    "t.columns =['gene', 'percent_cell', 'imp_sumup_norm']\n",
    "pearson_JGs = pearson_JGs.merge(t, how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "95c4cd03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>pearson_r</th>\n",
       "      <th>p</th>\n",
       "      <th>percent_cell</th>\n",
       "      <th>imp_sumup_norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ADORA2A</td>\n",
       "      <td>0.506924</td>\n",
       "      <td>2.928539e-176</td>\n",
       "      <td>0.082132</td>\n",
       "      <td>1.409286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GPRC5A</td>\n",
       "      <td>0.475163</td>\n",
       "      <td>3.867412e-152</td>\n",
       "      <td>0.001341</td>\n",
       "      <td>0.003646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RGS9</td>\n",
       "      <td>0.429748</td>\n",
       "      <td>8.539149e-122</td>\n",
       "      <td>0.193765</td>\n",
       "      <td>1.499032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DRD2</td>\n",
       "      <td>0.375064</td>\n",
       "      <td>6.179269e-91</td>\n",
       "      <td>0.084814</td>\n",
       "      <td>0.354933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TIMP1</td>\n",
       "      <td>0.329728</td>\n",
       "      <td>1.720298e-69</td>\n",
       "      <td>0.012068</td>\n",
       "      <td>0.009975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>EPS8L2</td>\n",
       "      <td>0.318834</td>\n",
       "      <td>7.518859e-65</td>\n",
       "      <td>0.142139</td>\n",
       "      <td>0.991074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>STARD8</td>\n",
       "      <td>0.312137</td>\n",
       "      <td>4.310113e-62</td>\n",
       "      <td>0.177338</td>\n",
       "      <td>1.546082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RAB37</td>\n",
       "      <td>0.301625</td>\n",
       "      <td>6.646402e-58</td>\n",
       "      <td>0.133087</td>\n",
       "      <td>0.518867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>C1QL2</td>\n",
       "      <td>0.296869</td>\n",
       "      <td>4.582274e-56</td>\n",
       "      <td>0.119008</td>\n",
       "      <td>1.695864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ABHD12B</td>\n",
       "      <td>0.283146</td>\n",
       "      <td>5.931402e-51</td>\n",
       "      <td>0.062018</td>\n",
       "      <td>0.537246</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene  pearson_r              p  percent_cell  imp_sumup_norm\n",
       "0  ADORA2A   0.506924  2.928539e-176      0.082132        1.409286\n",
       "1   GPRC5A   0.475163  3.867412e-152      0.001341        0.003646\n",
       "2     RGS9   0.429748  8.539149e-122      0.193765        1.499032\n",
       "3     DRD2   0.375064   6.179269e-91      0.084814        0.354933\n",
       "4    TIMP1   0.329728   1.720298e-69      0.012068        0.009975\n",
       "5   EPS8L2   0.318834   7.518859e-65      0.142139        0.991074\n",
       "6   STARD8   0.312137   4.310113e-62      0.177338        1.546082\n",
       "7    RAB37   0.301625   6.646402e-58      0.133087        0.518867\n",
       "8    C1QL2   0.296869   4.582274e-56      0.119008        1.695864\n",
       "9  ABHD12B   0.283146   5.931402e-51      0.062018        0.537246"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i=0\n",
    "#pearson_JGs.sort_values('pearson_r', ascending = False).to_csv('Transfer' + str(i+1) + '/T' + str(i+1) + '_stats.csv', index = None)\n",
    "pearson_JGs = pd.read_csv('Transfer' + str(i+1) + '/T' + str(i+1) + '_stats.csv')\n",
    "pearson_JGs.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9c34013a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0.4233642320041345, 0.0)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "select = ~(pearson_JGs['pearson_r'].isna() | pearson_JGs['imp_sumup_norm'].isna())\n",
    "stats.pearsonr(pearson_JGs['pearson_r'].values[select], pearson_JGs['imp_sumup_norm'].values[select])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2127dd4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "\n",
    "VizCus(pearson_JGs, x_name = 'imp_sumup_norm', x_label = 'Contribution to Location Prediction',\n",
    "       y_name = 'pearson_r', y_label = 'pearson_r', s = 14, c_name = 'percent_cell',\n",
    "       vmin = 0, vmax = 1, root = 'Transfer' + str(i+1) + '/',\n",
    "       save = 'Transfer' + str(i+1) + '_imp_r_pcell.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d69d868",
   "metadata": {},
   "source": [
    "# Visualize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "496a12f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>pearson_r</th>\n",
       "      <th>p</th>\n",
       "      <th>percent_cell</th>\n",
       "      <th>imp_sumup_norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ADORA2A</td>\n",
       "      <td>0.506924</td>\n",
       "      <td>2.928539e-176</td>\n",
       "      <td>0.082132</td>\n",
       "      <td>1.409286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RGS9</td>\n",
       "      <td>0.429748</td>\n",
       "      <td>8.539149e-122</td>\n",
       "      <td>0.193765</td>\n",
       "      <td>1.499032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DRD2</td>\n",
       "      <td>0.375064</td>\n",
       "      <td>6.179269e-91</td>\n",
       "      <td>0.084814</td>\n",
       "      <td>0.354933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>EPS8L2</td>\n",
       "      <td>0.318834</td>\n",
       "      <td>7.518859e-65</td>\n",
       "      <td>0.142139</td>\n",
       "      <td>0.991074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>STARD8</td>\n",
       "      <td>0.312137</td>\n",
       "      <td>4.310113e-62</td>\n",
       "      <td>0.177338</td>\n",
       "      <td>1.546082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RAB37</td>\n",
       "      <td>0.301625</td>\n",
       "      <td>6.646402e-58</td>\n",
       "      <td>0.133087</td>\n",
       "      <td>0.518867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>C1QL2</td>\n",
       "      <td>0.296869</td>\n",
       "      <td>4.582274e-56</td>\n",
       "      <td>0.119008</td>\n",
       "      <td>1.695864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ABHD12B</td>\n",
       "      <td>0.283146</td>\n",
       "      <td>5.931402e-51</td>\n",
       "      <td>0.062018</td>\n",
       "      <td>0.537246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LCT</td>\n",
       "      <td>0.282876</td>\n",
       "      <td>7.425375e-51</td>\n",
       "      <td>0.056990</td>\n",
       "      <td>0.526227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LHX9</td>\n",
       "      <td>0.259128</td>\n",
       "      <td>1.117889e-42</td>\n",
       "      <td>0.078780</td>\n",
       "      <td>0.776470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DRD1</td>\n",
       "      <td>0.258049</td>\n",
       "      <td>2.515170e-42</td>\n",
       "      <td>0.131411</td>\n",
       "      <td>0.834571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>OTOF</td>\n",
       "      <td>0.252609</td>\n",
       "      <td>1.414995e-40</td>\n",
       "      <td>0.132752</td>\n",
       "      <td>0.850377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CRLF1</td>\n",
       "      <td>0.251491</td>\n",
       "      <td>3.201553e-40</td>\n",
       "      <td>0.103922</td>\n",
       "      <td>0.779520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PATJ</td>\n",
       "      <td>0.246726</td>\n",
       "      <td>9.915856e-39</td>\n",
       "      <td>0.213879</td>\n",
       "      <td>0.892367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANO5</td>\n",
       "      <td>0.244577</td>\n",
       "      <td>4.554928e-38</td>\n",
       "      <td>0.082803</td>\n",
       "      <td>0.641501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GBX2</td>\n",
       "      <td>0.236911</td>\n",
       "      <td>9.284796e-36</td>\n",
       "      <td>0.076768</td>\n",
       "      <td>1.193049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROBO3</td>\n",
       "      <td>0.229981</td>\n",
       "      <td>9.663108e-34</td>\n",
       "      <td>0.134764</td>\n",
       "      <td>0.736271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SYT9</td>\n",
       "      <td>0.227672</td>\n",
       "      <td>4.394017e-33</td>\n",
       "      <td>0.182367</td>\n",
       "      <td>0.542034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>A830036E02RIK</td>\n",
       "      <td>0.226564</td>\n",
       "      <td>9.031189e-33</td>\n",
       "      <td>0.181361</td>\n",
       "      <td>0.608569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>IGFN1</td>\n",
       "      <td>0.225119</td>\n",
       "      <td>2.298648e-32</td>\n",
       "      <td>0.068723</td>\n",
       "      <td>0.436895</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene  pearson_r              p  percent_cell  imp_sumup_norm\n",
       "0         ADORA2A   0.506924  2.928539e-176      0.082132        1.409286\n",
       "2            RGS9   0.429748  8.539149e-122      0.193765        1.499032\n",
       "3            DRD2   0.375064   6.179269e-91      0.084814        0.354933\n",
       "5          EPS8L2   0.318834   7.518859e-65      0.142139        0.991074\n",
       "6          STARD8   0.312137   4.310113e-62      0.177338        1.546082\n",
       "7           RAB37   0.301625   6.646402e-58      0.133087        0.518867\n",
       "8           C1QL2   0.296869   4.582274e-56      0.119008        1.695864\n",
       "9         ABHD12B   0.283146   5.931402e-51      0.062018        0.537246\n",
       "10            LCT   0.282876   7.425375e-51      0.056990        0.526227\n",
       "11           LHX9   0.259128   1.117889e-42      0.078780        0.776470\n",
       "12           DRD1   0.258049   2.515170e-42      0.131411        0.834571\n",
       "13           OTOF   0.252609   1.414995e-40      0.132752        0.850377\n",
       "14          CRLF1   0.251491   3.201553e-40      0.103922        0.779520\n",
       "15           PATJ   0.246726   9.915856e-39      0.213879        0.892367\n",
       "16           ANO5   0.244577   4.554928e-38      0.082803        0.641501\n",
       "18           GBX2   0.236911   9.284796e-36      0.076768        1.193049\n",
       "20          ROBO3   0.229981   9.663108e-34      0.134764        0.736271\n",
       "21           SYT9   0.227672   4.394017e-33      0.182367        0.542034\n",
       "22  A830036E02RIK   0.226564   9.031189e-33      0.181361        0.608569\n",
       "25          IGFN1   0.225119   2.298648e-32      0.068723        0.436895"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GLs = pearson_JGs[pearson_JGs['percent_cell']>0.05].sort_values('pearson_r', ascending = False)['gene'][:20]\n",
    "pearson_JGs[pearson_JGs['percent_cell']>0.05].sort_values('pearson_r', ascending = False).head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "feca1c71",
   "metadata": {},
   "outputs": [],
   "source": [
    "cmap = sns.cubehelix_palette(n_colors = 32,start = 2, rot=1.5, as_cmap = True)\n",
    "\n",
    "GLs = pearson_JGs[pearson_JGs['percent_cell']>0.05].sort_values('pearson_r', ascending = False)['gene'][:20]\n",
    "    \n",
    "s1 = 6\n",
    "s2 = 54\n",
    "for gene in GLs:\n",
    "    print('*'*32)\n",
    "    print(gene)\n",
    "    print('*'*32)\n",
    "    print('scRNAseq:')\n",
    "    Vis.DrawGenes2(adata_query,coords_name = 'spatial_mapping', gene = gene, lim = False, cmap = cmap,\n",
    "                   FM = True, CTL = None, c_name = 'simp_name', root = 'Transfer' + str(i+1) + '/FM_Valid2/',\n",
    "                   s = s1, title = False, save = 'SC')\n",
    "    print('ST:')\n",
    "    Vis.DrawGenes2(adata_ref, coords_name = 'spatial', gene = gene, lim = False, cmap = cmap,\n",
    "                   FM = True, CTL = None, c_name = 'SSV2', root = 'Transfer' + str(i+1) + '/FM_Valid2/', \n",
    "                   s = s2, title = False, save = 'ST' + str(i+1))\n",
    "    print('Imputed ST:')\n",
    "    Vis.DrawGenes2(adata_impute, coords_name = 'spatial', gene = gene, lim = False, cmap = cmap,\n",
    "                   FM = True, CTL = None, c_name = 'SSV2', root = 'Transfer' + str(i+1) + '/FM_Valid2/', \n",
    "                   s = s2, title = False, save = 'ST' + str(i+1) + '_Impute')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2444c333",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3550195d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
